<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="author" content="Tade Souaiaia" />
      <link rel="shortcut icon" href="../../../img/favicon.ico" />
    <title>Day4b.docx - SibArc</title>
    <link rel="stylesheet" href="../../../css/theme.css" />
    <link rel="stylesheet" href="../../../css/theme_extra.css" />
        <link href="../../../css/extra.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Day4b.docx";
        var mkdocs_page_input_path = "tmp/practical_docs_hidden/Day4b.docx.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../../js/html5shiv.min.js"></script>
    <![endif]--> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../../.." class="icon icon-home"> SibArc
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../..">Home</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Quick Start</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../../quik_install/">Installation and Data</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../quik_demo/">Running SibArc</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../quik_result/">Interpreting Results</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">About</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../../about_future/">Future Work</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Misc</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../workshop_practical_paul.docx/">Day 1 (AM)</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../workshop_practical_tade/">Day 1 (PM)</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../workshop_practical_clive.docx/">Day 2 (AM)</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../..">SibArc</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../../.." class="icon icon-home" aria-label="Docs"></a></li>
      <li class="breadcrumb-item active">Day4b.docx</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/tadesouaiaia/sibArc-website/edit/master/docs/tmp/practical_docs_hidden/Day4b.docx.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h2 id="day-4-practical-2-introduction-to-admixture-analysis">Day 4 - Practical 2: Introduction to Admixture analysis</h2>
<h3 id="module-goals">Module Goals</h3>
<p>The goal of this practical is to provide you with basic understanding of Admixture and the basic elements behind Admixture PRS scores. Upon completion of this practical, you should:
* Gain familiarity with a variety of tools used in the context of admixed population research
* Go through the basic steps of formulating admixture-informed polygenic risk scores</p>
<h3 id="part1-plot-decay-of-ancestry-ld-over-time">Part1: Plot Decay of Ancestry LD over time</h3>
<p>Here we explore how Admixture LD varies over time and as a function of the genetic distance between loci.
In R:
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>library(ggplot2)
library(reshape2)
library(viridis)

#range of values of r (recombination fraction)
r = seq(0, 0.5, 0.1)
dtmat = matrix(NA, nrow = 6, ncol = 10) # matrix to store values of dt


for(i in 1:6){
  dt = 0.25
  for(j in 1:10){
    dtmat[i,j] = dt
    dt = dt*(1 -r[i])^j
  }
}

dtmat = reshape2::melt(dtmat)
colnames(dtmat) = c(&quot;r&quot;,&quot;g&quot;,&quot;Dt&quot;)
dtmat$r = (dtmat$r - 1)/10

ggplot(dtmat, 
       aes(x = g, y = Dt, group = r, color = as.factor(r))) +
  geom_line(linewidth = 1.2) +
  theme_minimal(base_size = 14) +
  theme(
    axis.text = element_text(size = 14),
    axis.title = element_text(size = 16),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12),
    plot.title = element_text(size = 18, face = &quot;bold&quot;),
    plot.subtitle = element_text(size = 14),
    panel.grid.major = element_line(color = &quot;gray80&quot;)
  ) +
  scale_color_viridis_d() +
  scale_x_continuous(breaks = c(1:10)) +
  labs(
    title = &quot;Decay of Linkage Disequilibrium Over Generations&quot;,
    subtitle = &quot;Effect of Recombination Fraction (r) on Admixture LD&quot;,
    x = &quot;Generations since admixture (t)&quot;,
    y = &quot;Admixture LD&quot;,
    color = &quot;Recombination Fraction (r)&quot;
  )
</code></pre></div></p>
<h4 id="questions"><strong>Questions</strong></h4>
<h5 id="i-describe-what-happens-to-admixture-ld-over-time">(i) Describe what happens to admixture LD over time?</h5>
<h5 id="ii-why-does-recombination-also-have-an-impact">(ii) Why does recombination also have an impact?</h5>
<h3 id="part-2-global-ancestry-inference">Part 2: Global Ancestry Inference</h3>
<p>We will now run an analysis using the software ADMIXTURE to calculate global ancestry proportions across a sample of 28 individuals. Here we perform a supervised analysis. Please execute the following code from location ~/RFMIX_WCS_2024/data/plink/
 <code>./admixture samples_n28_qc_thin.bed 2 --supervised -j4</code></p>
<h4 id="questions_1"><strong>Questions</strong></h4>
<h5 id="i-what-do-you-think-the-number-specified-after-the-inout-file-represents">(i) What do you think the number specified after the inout file represents?</h5>
<p>The next step is to create a plot the results
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code># Plot Global Ancestry Results
 R
 library(ggplot2)
 library(reshape2)

 # Read the data from files
 fam &lt;- read.table(&quot;samples_n28_qc_thin.fam&quot;, header = FALSE)
 pop &lt;- read.table(&quot;samples_n28_qc_thin.pop&quot;, header = FALSE)
 Q &lt;- read.table(&quot;samples_n28_qc_thin.2.Q&quot;, header = FALSE)

 # Merge the data into one data frame
 merged_data &lt;- cbind(fam, pop, Q)

 # Extract necessary columns (assuming the structure of the files as in the image)
 colnames(merged_data) &lt;- c(&quot;FID&quot;, &quot;IID&quot;, &quot;PaternalID&quot;, &quot;MaternalID&quot;, &quot;Sex&quot;, &quot;Phenotype&quot;, &quot;Population&quot;, &quot;AFR&quot;, &quot;EUR&quot;)

 # Order data by decreasing AFR
 merged_data &lt;- merged_data[order(-merged_data$AFR), ]

 # Prepare data for plotting
 plot_data &lt;- merged_data[, c(&quot;IID&quot;, &quot;AFR&quot;, &quot;EUR&quot;)]
 plot_data$IID &lt;- factor(plot_data$IID, levels = plot_data$IID)  # Ensure order is maintained in plot
 plot_data_melted &lt;- melt(plot_data, id.vars = &quot;IID&quot;)

 # Create the plot
 p &lt;- ggplot(plot_data_melted, aes(x = IID, y = value, fill = variable)) +
   geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
   scale_fill_manual(values = c(&quot;AFR&quot; = &quot;red&quot;, &quot;EUR&quot; = &quot;darkgreen&quot;)) +
   labs(x = &quot;Individual ID&quot;, y = &quot;Ancestry Proportion&quot;, fill = &quot;Ancestry&quot;, title = &quot;Global Ancestry Proportions&quot;) +
   theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 10, face = &quot;bold&quot;),
         axis.text.y = element_text(face = &quot;bold&quot;),  # Bold y-axis numbers
         axis.title.x = element_text(face = &quot;bold&quot;),
         axis.title.y = element_text(face = &quot;bold&quot;),
         legend.title = element_text(face = &quot;bold&quot;),
         plot.title = element_text(face = &quot;bold&quot;, hjust = 0.5))

 # Save the plot with specified dimensions
 ggsave(&quot;ancestry_plot.png&quot;, plot = p, width = 10, height = 6, units = &quot;in&quot;, dpi = 300)
</code></pre></div></p>
<h4 id="questions_2">Questions</h4>
<h5 id="i-what-are-the-ancestry-assignments-of-the-28-individuals-provide-estimated-proportions-where-necessary">(i) What are the ancestry assignments of the 28 individuals? (Provide estimated proportions where necessary)</h5>
<h3 id="part-3-local-ancestry-inference">Part 3: Local Ancestry Inference</h3>
<p>Next we will use the RFMix software to calculate local ancestry on chromosome 22 for the same 28 individuals. The RFMix algorithm uses an unsupervised learning algorithm.
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code># Run the following UNIX command from the home directory
module load cmake
module load bcftools/1.10.2

for i in {22..22}; do  
./software/rfmix -f ./data/rfmix/chr1-22_phased.bcf.gz -r ./reference/chr1-22.b.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.bcf.gz --analyze-range=26.86-31.80 -m ./data/rfmix/1KG_superpop_vs_ID.txt --chromosome=${i} -g ./reference/1kg_chr1-22.gmap --n-threads=4 -o ./out/rfmix/chr${i}.local_ancestry; done
</code></pre></div></p>
<h4 id="questions_3">Questions</h4>
<h5 id="i-what-information-is-provided-in-the-simulation-results-table-displayed-on-screen">(i) What information is provided in the simulation results table displayed on-screen ?</h5>
<h3 id="part-4-plot-local-admixture-on-chromosome-22">Part 4: Plot local admixture on chromosome 22</h3>
<p>In this step we will plot the output from the previous RFMix run. Execute the following code from the home directory</p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>R
library(ggplot2)
library(dplyr)
library(tidyr)

# Function to read the .msp.tsv file
read_msp_file &lt;- function(msp_file) {
  # Read the entire file as text
  lines &lt;- readLines(msp_file)

  # Determine the number of header lines (lines starting with &#39;#&#39;)
  header_lines &lt;- lines[grepl(&quot;^#&quot;, lines)]
  num_header_lines &lt;- length(header_lines)

  # Extract column names from the last header line
  col_names &lt;- strsplit(header_lines[num_header_lines], &quot;\t&quot;)[[1]]

  # Read the data skipping the header lines
  msp_df &lt;- read.table(msp_file, header = FALSE, skip = num_header_lines, sep = &quot;\t&quot;)

  # Assign column names
  colnames(msp_df) &lt;- col_names
  return(msp_df)
}

# Function to read the .Q file
read_Q_file &lt;- function(Q_file) {
  Q_df &lt;- read.table(Q_file, header = TRUE, sep = &quot;\t&quot;, comment.char = &quot;#&quot;)
  colnames(Q_df) &lt;- c(&quot;sample&quot;, &quot;AFR&quot;, &quot;AMR&quot;, &quot;EAS&quot;, &quot;EUR&quot;, &quot;SAS&quot;)
  return(Q_df)
}

# Specification of file paths
msp_file &lt;- &#39;./out/rfmix/chr22.local_ancestry.msp.tsv&#39;
Q_file &lt;- &#39;./out/rfmix/chr22.local_ancestry.rfmix.Q&#39;

# Read in files
msp_df &lt;- read_msp_file(msp_file)
Q_df &lt;- read_Q_file(Q_file)

# Check for NA or empty column names
print(colnames(msp_df))

# Extract ancestry columns based on pattern recognition of column names
ancestry_cols &lt;- colnames(msp_df)[grep(&quot;\\.0$|\\.1$&quot;, colnames(msp_df))]

# Function to determine ancestry based on RFMix codes
determine_ancestry &lt;- function(value) {
  ancestry_map &lt;- c(&quot;AFR&quot;, &quot;AMR&quot;, &quot;EAS&quot;, &quot;EUR&quot;, &quot;SAS&quot;)
  return(ancestry_map[value + 1])
}

# Rename columns uniquely
colnames(msp_df)[7:62] &lt;- paste0(&quot;haplo_&quot;, colnames(msp_df)[7:62])

# Prepare data for plotting
plot_data &lt;- msp_df %&gt;%
  pivot_longer(cols = starts_with(&quot;haplo_&quot;), names_to = &quot;haplo_col&quot;, values_to = &quot;ancestry_code&quot;, names_repair = &quot;unique&quot;) %&gt;%
  mutate(
    individual = gsub(&quot;\\..*&quot;, &quot;&quot;, haplo_col),
    strand = ifelse(grepl(&quot;\\.0$&quot;, haplo_col), &quot;Strand 1&quot;, &quot;Strand 2&quot;),
    ancestry = determine_ancestry(ancestry_code)
  )

# Create plot of chromosome 22 across the sample
plot &lt;- ggplot(plot_data, aes(x = sgpos, xend = egpos, y = interaction(individual, strand, lex.order = TRUE), yend = interaction(individual, strand, lex.order = TRUE), color = ancestry)) +
  geom_segment(linewidth = 4) +
  scale_color_manual(values = c(&quot;AFR&quot; = &quot;blue&quot;, &quot;AMR&quot; = &quot;orange&quot;, &quot;EAS&quot; = &quot;green&quot;, &quot;EUR&quot; = &quot;red&quot;, &quot;SAS&quot; = &quot;purple&quot;)) +
  labs(x = &quot;Genetic position (cM)&quot;, y = &quot;Individual ID and Haplotype&quot;, title = &quot;Local Ancestry Across Chromosome 22&quot;, color = &quot;Ancestry&quot;) +
  theme_minimal() +
  theme(axis.text.y = element_text(size = 8), axis.title.y = element_text(size = 10), axis.title.x = element_text(size = 10))

# Save the plot to a file
ggsave(&quot;./out/rfmix/local_ancestry_chromosome22_5Mb_subregion.png&quot;, plot = plot, width = 10, height = 8)
</code></pre></div>
<h4 id="questions_4">Questions</h4>
<h5 id="i-how-many-different-continental-ancestries-do-you-see-represented-across-the-58-strands">(i) How many different continental ancestries do you see represented across the 58 strands?</h5>
<h5 id="ii-why-is-the-number-of-different-ancestry-backgrounds-higher-than-it-was-in-the-previous-step">(ii) Why is the number of different ancestry backgrounds higher than it was in the previous step?</h5>
<h3 id="part-5-formatting-of-admixture-files-for-analysis-using-prsice">Part 5: Formatting of admixture files for analysis using PRSice</h3>
<h4 id="step-1-convert-phased-genotypes-to-genomicrange-format">Step 1 - Convert phased genotypes to GenomicRange format</h4>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code># Run from the home directory

library(vcfR)
library(memuse)
library(panelr)
library(GenomicRanges)
library(data.table)

clean_memory &lt;- function(vars_to_remove) {
  rm(list = vars_to_remove)
  gc()
}

# Read in phased VCF file
vcfr_inputfile_chr22 &lt;- read.vcfR(&quot;./data/rfmix/chr22_phased.vcf.gz&quot;, verbose = FALSE)
extracted_haps22 &lt;- extract.haps(vcfr_inputfile_chr22, mask = FALSE, unphased_as_NA = TRUE, verbose = TRUE)
extracted_snp_info22 &lt;- getFIX(vcfr_inputfile_chr22)

# Convert haplotypes to data frame
haps_df &lt;- data.frame(extracted_haps22, check.names = FALSE)
haps_dt &lt;- setDT(haps_df, keep.rownames = &quot;snps&quot;)

# Convert SNP info to data frame
snp_info_df &lt;- data.frame(extracted_snp_info22)

# Check for numerical and ordering consistency between haplotypes and SNP info
if (!identical(haps_dt[[&#39;snps&#39;]], snp_info_df[[&#39;ID&#39;]])) {
  stop(&quot;Numerical and ordering inconsistency between haplotypes and SNP info&quot;)
}


# Merge haplotypes and SNP info
merge_chr22 &lt;- cbind(snp_info_df, haps_dt)

# Convert to long format using panelr
chr22_haplo_long &lt;- long_panel(merge_chr22, prefix = &quot;_&quot;, begin = 0, end = 1, label_location = &quot;end&quot;, as_panel_data = FALSE)

# Insert &#39;end&#39; column after &#39;POS&#39; and create &#39;wave&#39; column
chr22_haplo_long$end &lt;- chr22_haplo_long$POS
chr22_haplo_long$wave &lt;- ifelse(chr22_haplo_long$wave == &quot;0&quot;, &quot;+&quot;, ifelse(chr22_haplo_long$wave == &quot;1&quot;, &quot;-&quot;, &quot;Z&quot;))

# Remove row names and drop redundant columns using base R
rownames(chr22_haplo_long) &lt;- NULL
drops &lt;- c(&quot;id&quot;, &quot;snps&quot;, &quot;QUAL&quot;, &quot;FILTER&quot;)
chr22_haplo_long &lt;- chr22_haplo_long[, !names(chr22_haplo_long) %in% drops]

# Rename columns
names(chr22_haplo_long)[3] &lt;- &quot;start&quot;
names(chr22_haplo_long)[1] &lt;- &quot;strand&quot;
header &lt;- gsub(&quot;.*_&quot;, &quot;&quot;, colnames(chr22_haplo_long)[7:ncol(chr22_haplo_long)])
names(chr22_haplo_long)[7:ncol(chr22_haplo_long)] &lt;- header

# Create GRanges object
gr_obj_chr22 &lt;- makeGRangesFromDataFrame(chr22_haplo_long, ignore.strand = FALSE, keep.extra.columns = TRUE)

# Save the GRanges object
saveRDS(gr_obj_chr22, file = &quot;./out/rfmix/chr22_phased_gr.rds&quot;)

# Clean up memory
clean_memory(c(&quot;vcfr_inputfile_chr22&quot;, &quot;extracted_haps22&quot;, &quot;merge_chr22&quot;, &quot;haps_df&quot;, &quot;haps_dt&quot;, &quot;snp_info_df&quot;, &quot;chr22_haplo_long&quot;, &quot;gr_obj_chr22&quot;))
</code></pre></div>
<h3 id="part-5-step-2-merge-genotypes-from-step-1-with-local-ancestry-calls-by-rfmix">Part 5: Step 2 - Merge genotypes from Step 1 with local ancestry calls by RFMix</h3>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code># Read in MSP file
msp &lt;- fread(&quot;./out/rfmix/chr22.local_ancestry.msp.tsv&quot;)

# Reformat genome-wide local ancestry output as GRanges object
colnames(msp)[1] &lt;- &quot;chm&quot;
msp_gr &lt;- makeGRangesFromDataFrame(msp,
                                   seqnames.field = &quot;chm&quot;,
                                   start.field = &quot;spos&quot;,
                                   end.field = &quot;epos&quot;,
                                   keep.extra.columns = TRUE)

# Convert the elements of the GRange object into a dataframe
chr22_rf &lt;- data.frame(seqnames = seqnames(msp_gr),
                       ranges = ranges(msp_gr),
                       strand = strand(msp_gr),
                       mcols(msp_gr), check.names = FALSE)
names(chr22_rf)[1:5] &lt;- c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;width&quot;, &quot;strand&quot;)
rownames(chr22_rf) &lt;- NULL

# Clean up column headers
header &lt;- gsub(&quot;.*_&quot;, &quot;&quot;, colnames(chr22_rf)[9:ncol(chr22_rf)])
names(chr22_rf)[9:ncol(chr22_rf)] &lt;- header

# Convert local ancestry calls from wide to long format
chr22_rf_long &lt;- long_panel(chr22_rf, prefix = &quot;.&quot;, begin = 0, end = 1, label_location = &quot;end&quot;, as_panel_data = FALSE)
chr22_rf_long &lt;- as.data.frame(chr22_rf_long, check.names = FALSE)
chr22_rf_long$strand &lt;- ifelse(chr22_rf_long$wave == &quot;0&quot;, &quot;+&quot;, ifelse(chr22_rf_long$wave == &quot;1&quot;, &quot;-&quot;, &quot;Z&quot;))

# Drop redundant columns
drops &lt;- c(&quot;wave&quot;, &quot;id&quot;)
chr22_rf_long &lt;- chr22_rf_long[, !(names(chr22_rf_long) %in% drops)]
rownames(chr22_rf_long) &lt;- NULL

# Convert the reconfigured dataframe file into a GRanges object
chr22_msp_gr &lt;- makeGRangesFromDataFrame(chr22_rf_long, ignore.strand = FALSE, keep.extra.columns = TRUE)

# Ensure chr22_haplo_long is a GRanges object before finding overlaps
chr22_haplo_gr &lt;- makeGRangesFromDataFrame(chr22_haplo_long, ignore.strand = FALSE, keep.extra.columns = TRUE)

# Find overlaps and store matching features
# Matching is coordinates based. Base position (&quot;start&quot;/&quot;end&quot;)is used to match the two files
matched_regions &lt;- findOverlaps(chr22_haplo_gr, chr22_msp_gr)
chr22_haps_lanc_gr &lt;- chr22_haplo_gr[queryHits(matched_regions)]  #Store matching features in a new dataframe, add metadata from RFmix output.
mcols(chr22_haps_lanc_gr) &lt;- cbind.data.frame(mcols(chr22_haps_lanc_gr), mcols(chr22_msp_gr[subjectHits(matched_regions)]))

# Local ancestry calls are now aligned with genotypic data and positional coordinates
# Convert to dataframe without adding &#39;X&#39; to numeric column names
genes_df &lt;- as.data.frame(chr22_haps_lanc_gr)
names(genes_df) &lt;- sub(&#39;^X&#39;, &#39;&#39;, names(genes_df))

# Output the dataframe
write.table(genes_df, file = &quot;./out/rfmix/chr22_phased_geno_lanc.txt&quot;, quote = FALSE, row.names=F)

# Clean up memory
rm(list = c(&quot;chr22_haplo_gr&quot;, &quot;chr22_haplo_long&quot;, &quot;msp&quot;, &quot;msp_gr&quot;, &quot;chr22_rf&quot;, &quot;header&quot;,
            &quot;chr22_rf_long&quot;, &quot;chr22_msp_gr&quot;, &quot;matched_regions&quot;,
            &quot;chr22_haps_lanc_gr&quot;))
gc()
</code></pre></div>
<h4 id="part-5-step-3-create-separate-plink-files">Part 5: Step 3 - Create separate Plink files</h4>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code># Partition genotype and local ancestry data
chr22_genos &lt;- genes_df[, c(1, 2, 6, 9:36)]
chr22_LA &lt;- genes_df[, c(1, 2, 6, 40:ncol(genes_df))]

# Reintegrate in interleaved format
d &lt;- chr22_genos[, -c(1:3)]  # Retain ID columns only
d2 &lt;- chr22_LA[, -c(1:3)]

# Prepare headers
indx &lt;- rbind(names(d), names(d2))
dmerge &lt;- cbind(d, d2)
dfinal &lt;- dmerge[, indx]

# Convert LAnc data from long to wide format
LA_wide &lt;- lapply(1:ncol(chr22_LA), function(i) as.data.frame(matrix(chr22_LA[, i], ncol = 2, byrow = TRUE)))

# Check length of LA_wide
length(LA_wide)  # Expected number. Each item contains a set of matched strand pairs per individual

LA_final &lt;- as.data.frame(do.call(cbind, LA_wide))
LA_final &lt;- LA_final[, -c(2, 4, 6)]  # Remove duplicates of first 3 cols
colnames(LA_final)[4:ncol(LA_final)] &lt;- colnames(dfinal)  # Apply headers

# Convert genotype calls to horizontal orientation
geno_wide &lt;- lapply(1:ncol(chr22_genos), function(i) as.data.frame(matrix(chr22_genos[, i], ncol = 2, byrow = TRUE)))  # Genotypic part: Get horizontal genotypes
geno_final &lt;- as.data.frame(do.call(cbind, geno_wide))  # Merge horizontal genotypes

# Clean up and finalize geno_final
geno_final &lt;- geno_final[, -c(2, 4, 6)]  # Redundant columns
colnames(geno_final)[4:ncol(geno_final)] &lt;- colnames(dfinal)

# Name columns
colnames(geno_final)[1:3] &lt;- c(&quot;CHROM&quot;, &quot;BP&quot;, &quot;ID&quot;)
colnames(LA_final)[1:3] &lt;- c(&quot;CHROM&quot;, &quot;BP&quot;, &quot;ID&quot;)

# Write final tables
write.table(geno_final, &quot;./out/rfmix/chr22_geno.txt&quot;, row.names = FALSE, quote = FALSE)
write.table(LA_final, &quot;./out/rfmix/chr22_LA.txt&quot;, row.names = FALSE, quote = FALSE)
</code></pre></div>
<h4 id="part-5-step-4-use-custom-software-rftransform-to-create-plink-files-for-input-into-prsice">Part 5: Step 4 - Use custom software (RFTransform) to create Plink files for input into PRSice</h4>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>module load cmake
./software/RFTransform/build/RFTransformer ./out/rfmix/chr22_geno.txt ./out/rfmix/chr22_LA.txt ./out/plink/chr22

# Perform general QC ahead of running PRSice on ancestry-deconvolved individuals
# (i) Remove SNPs with low minor allele count (MAC)

# Plink - Remove monomorphic SNPs (minor allele count 0-4).
#AFR
for i in {22..22}; do
        ./software/plink2 \
        --bfile ./out/plink/chr${i}-AFR \
        --mac 5 \
        --make-bed \
        --out ./out/plink/chr${i}-AFR.mac
done

#EUR
for i in {22..22}; do
        ./software/plink2 \
        --bfile ./out/plink/chr${i}-EUR \
        --mac 5 \
        --make-bed \
        --out ./out/plink//chr${i}-EUR.mac
done

# PRSice - Generate ancestry-specific weights

# AFR
Rscript ./software/PRSice.R \
--prsice ./software/PRSice_linux \
--base ./data/plink/AFR-BMI.Phenotype.glm.linear \
--extract ./data/plink/snp.valid \
--A1 A1 \
--pvalue P \
--stat BETA \
--pheno ./data/plink/pheno.plink \
--beta \
--snp ID \
--score sum \
--target ./out/plink/chr22-AFR.mac \
--binary-target F \
--out ./out/prsice/BMI_AFR-base

# EUR
Rscript ./software/PRSice.R \
--prsice ./software/PRSice_linux \
--base ./data/plink/EUR-BMI.Phenotype.glm.linear \
--extract ./data/plink/snp.valid \
--A1 A1 \
--pvalue P \
--stat BETA \
--pheno ./data/plink/pheno.plink \
--beta \
--snp ID \
--score sum \
--target ./out/plink/chr22-EUR.mac \
--binary-target F \
--out ./out/prsice/BMI_EUR-base
</code></pre></div>
<h4 id="part-5-step-5-evaluate-the-admixture-informed-prs">Part 5: Step 5 - Evaluate the Admixture-informed PRS</h4>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>library(dplyr)

# Read the PRS files into dataframes
file1 &lt;- read.table(&quot;./out/prsice/BMI_EUR-base.best&quot;, header = TRUE, check.names=F)
file2 &lt;- read.table(&quot;./out/prsice/BMI_AFR-base.best&quot;, header = TRUE, check.names=F)

# Add the fourth column of both files
file1$PRS_SUM &lt;- file1$PRS + file2$PRS

# Load the phenotype data
pheno &lt;- read.table(&quot;./data/plink/pheno.plink&quot;, header = F)
colnames(pheno) &lt;- c(&quot;FID&quot;, &quot;IID&quot;, &quot;phenotype&quot;)

# Convert phenotype column to numeric
pheno$phenotype &lt;- as.numeric(pheno$phenotype)

# Merge PRS data with phenotype data
merged_data &lt;- merge(file1[, c(&quot;FID&quot;, &quot;IID&quot;, &quot;PRS&quot;, &quot;PRS_SUM&quot;)], pheno, by = c(&quot;FID&quot;, &quot;IID&quot;))
merged_data &lt;- merge(merged_data, file2[, c(&quot;FID&quot;, &quot;IID&quot;, &quot;PRS&quot;)], by = c(&quot;FID&quot;, &quot;IID&quot;))

# Rename columns
names(merged_data)[names(merged_data) == &quot;PRS.x&quot;] &lt;- &quot;PRS1&quot;
names(merged_data)[names(merged_data) == &quot;PRS.y&quot;] &lt;- &quot;PRS2&quot;

# Perform linear regression for each PRS and the combined PRS
model1 &lt;- lm(phenotype ~ PRS1, data = merged_data)
model2 &lt;- lm(phenotype ~ PRS2, data = merged_data)
model_sum &lt;- lm(phenotype ~ PRS_SUM, data = merged_data)

# Extract R-squared values
r_squared1 &lt;- summary(model1)$r.squared
r_squared2 &lt;- summary(model2)$r.squared
r_squared_sum &lt;- summary(model_sum)$r.squared

# Print the R-squared values
cat(&quot;R-squared for PRS1: &quot;, r_squared1, &quot;\n&quot;)
cat(&quot;R-squared for PRS2: &quot;, r_squared2, &quot;\n&quot;)
cat(&quot;R-squared for PRS_SUM: &quot;, r_squared_sum, &quot;\n&quot;)
</code></pre></div>
<h4 id="questions_5"><strong>Questions</strong></h4>
<h5 id="i-how-does-the-r-squared-of-the-combined-ancestry-prs-perform-relative-to-the-2-partial-genome-prss">(i) How does the R-squared of the combined-ancestry PRS perform relative to the 2 partial-genome PRSs?</h5>
              
            </div>
          </div><footer>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/tadesouaiaia/sibArc-website" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
    
  </span>
</div>
    <script src="../../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../../..";</script>
    <script src="../../../js/theme_extra.js"></script>
    <script src="../../../js/theme.js"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      <script src="../../../javascripts/details.js"></script>
      <script src="../../../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
