\documentclass[12pt,openany]{scrbook}

\usepackage{booktabs} % Nicer table
\usepackage[inline]{enumitem} % Allow inline enumeriate
\usepackage{fancyhdr} % Nicer page format
\usepackage{float} % allow better location setup for tables and figure
\usepackage[T1]{fontenc} % allow copy and paste the words
\usepackage{geometry} % Something to arrange the page
\usepackage{cmap} % Make the pdf searchable
% define xcolor early on to avoid clash
\usepackage[table, dvipsnames]{xcolor}  % define colors
\usepackage[hidelinks]{hyperref} % Nicer hyper link
\usepackage{listings} % For printing computer codes
\usepackage{etoolbox} % For condition
\usepackage{mdframed} % This is require to generate the nice boxes
\usepackage{ragged2e} % For text alignment
\usepackage{tikz} % to  beautify the key stroke with shadows
\usetikzlibrary{shadows}
\usepackage{pdfpages} % Allow incorporation of a separate pdf
\usepackage[acronym,nomain,toc,makeindex ]{glossaries} % For making the glossaries
\usepackage{setspace} % Allow changing the spacing
\usepackage{textcomp} % This and inputenc is required to display '
\usepackage{subcaption} % For subfigures
\usepackage[utf8]{inputenc} 
% Package for code style
\usepackage{accsupp}% http://ctan.org/pkg/accsupp
%\newcommand{\emptyaccsupp}[1]{\BeginAccSupp{ActualText={}}#1\EndAccSupp{}}
\newcommand\emptyaccsupp[1]{\BeginAccSupp{ActualText={}}#1\EndAccSupp{}}
\usepackage{cleveref} % Smart reference, need to load after other packages
\usepackage{subfiles} % For including other latex file 
\usepackage{silence}
\WarningFilter{scrbook}{Usage of package `fancyhdr'} % Remove the unnecessary warning
\usepackage[hyperref=true,maxcitenames=3,maxbibnames=3,url=false,doi=false,isbn=false,backref=true,natbib=true,backend=biber,sorting=nyvt,defernumbers=false, style=authoryear]{biblatex}
%
% Referencing stuff
%
\addbibresource[datatype=bibtex]{citations.bib}

\newcolumntype{P}[1]{>{\RaggedRight\hspace{0pt}}p{#1}} % Define new column type for vertical alignment 
%
% Generate the folder list
%


\usepackage{forest}

\definecolor{folderbg}{RGB}{124,166,198}
\definecolor{folderborder}{RGB}{110,144,169}

\def\Size{4pt}
\tikzset{
	folder/.pic={
		\filldraw[draw=folderborder,top color=folderbg!50,bottom color=folderbg]
		(-1.05*\Size,0.2\Size+5pt) rectangle ++(.75*\Size,-0.2\Size-5pt);  
		\filldraw[draw=folderborder,top color=folderbg!50,bottom color=folderbg]
		(-1.15*\Size,-\Size) rectangle (1.15*\Size,\Size);
	}
}
% glossaries
\makeglossaries
\newacronym{snp}{SNP}{Single Nucleotide Polymorphism}
\newacronym{ld}{LD}{Linkage Disequilibrium}
\newacronym[longplural={Genome Wide Association Studies},plural={GWAS}, \glsshortpluralkey={GWAS}]{gwas}{GWAS}{Genome-Wide Association Study}
\newacronym{prs}{PRS}{Polygenic Risk Score}
\newacronym{pc}{PC}{Principal Component}
\newacronym{cad}{CAD}{Coronary artery disease}
\newacronym{or}{OR}{Odd Ratios}
\newacronym{msigdb}{MSigDB}{Molecular Signatures Database}
\newacronym{gmt}{GMT}{Gene Matrix Transposed file format}
\newacronym{gtf}{GTF}{General Transfer Format}
\newacronym{bed}{BED}{Browser Extensible Data}
% Customize the code styles
% Color of the code box
\newtoggle{windows}
\newtoggle{mac}
\global\toggletrue{windows}
\global\togglefalse{mac}

\makeatletter
\newcommand{\iftoggleverb}[1]{%
	\ifcsdef{etb@tgl@#1}
	{\csname etb@tgl@#1\endcsname\iftrue\iffalse}
	{\etb@noglobal\etb@err@notoggle{#1}\iffalse}%
}
\makeatother


%\toggletrue{windows}
\newtoggle{printing}
\toggletrue{printing}
\iftoggle{printing}{%
	\definecolor{backcolour}{rgb}{1,1,1}
	\definecolor{codegreen}{rgb}{0.3,0.3,0.3}
	\definecolor{codegray}{rgb}{0,0,0}
	\definecolor{codepurple}{rgb}{0,0,0}
	\definecolor{codemagenta}{rgb}{0,0,0}
	\definecolor{warningshade}{rgb}{0.86,0.86,0.86}
	\definecolor{questionshade}{rgb}{1,1,1}
	\definecolor{darkgreen}{rgb}{0.09, 0.45, 0.27}
	\lstset{frame=single}
}{%
	\definecolor{backcolour}{rgb}{0.95,0.95,0.92}
	\definecolor{codegreen}{rgb}{0,0.6,0}
	\definecolor{codegray}{rgb}{0.5,0.5,0.5}
	\definecolor{codepurple}{rgb}{0.58,0,0.82}
	\definecolor{codemagenta}{rgb}{1,0,1}
	\definecolor{warningshade}{rgb}{1,0.85,0.85}
	\definecolor{questionshade}{rgb}{1,0.95,0.85}
	\definecolor{darkgreen}{rgb}{0.09, 0.45, 0.27}
}

\renewcommand{\ttdefault}{cmtt}
\makeatletter
\lst@CCPutMacro\lst@ProcessOther {"2D}{\lst@ttfamily{-{}}{-{}}}
\@empty\z@\@empty
\makeatother
\lstdefinestyle{codestyle}{
	columns=fullflexible,
	upquote=true, 
	aboveskip=5pt,
	belowskip=10pt,
	basicstyle=\tiny\ttfamily, 
	numbers=left,                    % where to put the line-numbers
	numberstyle=\tiny\color{codegray}\emptyaccsupp,
	stepnumber=1,
	numbersep=13pt, 
	xleftmargin=20pt,
	xrightmargin=10pt,
	framesep=5pt,
	framerule=3pt,
	frame=leftline, 
	rulecolor=\color{darkgreen},  
	backgroundcolor=\color{backcolour},   
	commentstyle=\color{codegreen},
	keywordstyle=\color{codemagenta},
	stringstyle=\color{codepurple},
	basicstyle=\footnotesize,
	breakatwhitespace=true,         
	breaklines=true,                 
	captionpos=b,                    
	keepspaces=true,                      
	showspaces=false,                
	showstringspaces=false,
	showtabs=false,                  
	tabsize=2,
	prebreak=\small\symbol{'134},
}
\lstset{style=codestyle}
\lstset{emph={%  
		Rscript.exe, Rscript%
	},emphstyle={\color{codemagenta}}%
}

% Command to generate the keystroke picture
\newcommand*\keystroke[1]{%
	\tikz[baseline=(key.base)]
	\node[%
	draw,
	fill=white,
	drop shadow={shadow xshift=0.25ex,shadow yshift=-0.25ex,fill=black,opacity=0.75},
	rectangle,
	rounded corners=2pt,
	inner sep=1pt,
	line width=0.5pt,
	font=\scriptsize\sffamily
	](key) {#1\strut}
	;
}
% add the mac cmd key for printing
\DeclareRobustCommand{\cmdkey}{\raisebox{-.035em}{\includegraphics[height=.75em]{img/command}}}
% add the windows flag key for printing
\DeclareRobustCommand{\faWindows}{\raisebox{-.035em}{\includegraphics[height=.75em]{img/windows.jpg}}}

% Define our question boxes' coloring and dimension

\global\mdfdefinestyle{notice}{
	leftmargin=0pt,
	skipabove=5pt,
	rightmargin=0pt,
	skipbelow=5pt,
	innermargin=0pt,
	outermargin=0pt,
	innerleftmargin=5pt,
	innerrightmargin=5pt,
	innertopmargin=5pt,
	innerbottommargin=5pt,
	leftline=true,
	topline=true,
	rightline=true,
	bottomline=true,
	linecolor=black,
	linewidth=1pt,
	roundcorner=5pt,
}

\global\mdfdefinestyle{warningbox}{
	leftmargin=0pt,
	skipabove=5pt,
	rightmargin=0pt,
	skipbelow=5pt,
	%splittopskip=0pt,
	%splitbottomskip=0pt,
	innermargin=0pt,
	outermargin=0pt,
	innerleftmargin=0pt,
	innerrightmargin=0pt,
	innertopmargin=0pt,
	innerbottommargin=0pt,
	leftline=false,
	topline=false,
	rightline=false,
	bottomline=false,
	linecolor=black,
	linewidth=1pt,
	roundcorner=5pt,
	backgroundcolor=warningshade,
	leftline=true,
	topline=true,
	rightline=true,
	bottomline=true,
	innerleftmargin=5pt,
	innerrightmargin=5pt,
	innertopmargin=5pt,
	innerbottommargin=5pt,
	nobreak=true,
}
\global\mdfdefinestyle{questionsbox}{
	leftmargin=0pt,
	skipabove=5pt,
	rightmargin=0pt,
	skipbelow=5pt,
	innermargin=0pt,
	outermargin=0pt,
	innerleftmargin=0pt,
	innerrightmargin=0pt,
	innertopmargin=0pt,
	innerbottommargin=0pt,
	leftline=false,
	topline=false,
	rightline=false,
	bottomline=false,
	linecolor=black,
	linewidth=1pt,
	roundcorner=5pt,
	backgroundcolor=questionshade,
	leftline=true,
	topline=true,
	rightline=true,
	bottomline=true,
	innerleftmargin=10pt,
	innerrightmargin=10pt,
	innertopmargin=5pt,
	innerbottommargin=5pt,
	nobreak=true,
}

\newlength{\iconspacinglength}
\setlength{\iconspacinglength}{0.5cm}
\newlength{\questionspacing}
\setlength{\questionspacing}{1cm}
\newenvironment{questions}{%
	\begin{mdframed}[style=questionsbox]%
		\setlength{\parskip}{\questionspacing}%
		\setlength{\parindent}{0pt}%
		\addtolength{\iconspacinglength}{\mdflength{leftmargin}}%
		\addtolength{\iconspacinglength}{\mdflength{innerleftmargin}}%
		\makebox[0pt][r]{\smash{\raisebox{-.25\height}{%
					\includegraphics[height=1cm]{./img/questions.png}%
					\hspace{\iconspacinglength}%
		}}}\ignorespaces%
	}%
	{%
		\vspace{\questionspacing}%
	\end{mdframed}%
}
\newenvironment{warning}{%
	\begin{mdframed}[style=warningbox]%
		\makebox[0pt][r]{\smash{\raisebox{-.6\height}{%
					\includegraphics[height=1cm]{./img/warning.png}%
					\hspace{\iconspacinglength}%
		}}}\ignorespaces%
	}%
	{%
	\end{mdframed}%
}


\newenvironment{information}{%
	\begin{mdframed}[style=notice]%
		%\addtolength{\iconspacinglength}{\mdflength{leftmargin}}%
		%\addtolength{\iconspacinglength}{\mdflength{innerleftmargin}}%
		\makebox[0pt][r]{\smash{\raisebox{-.6\height}{%
					\includegraphics[height=1cm]{./img/info.png}%
					\hspace{\iconspacinglength}%
		}}}\ignorespaces%
	}%
	{%
	\end{mdframed}%
}


\newenvironment{note}
{%
	\begin{mdframed}[style=notice]%
		\makebox[0pt][r]{\smash{\raisebox{-.6\height}{%
					\includegraphics[height=1cm]{./img/notes.png}%
					\hspace{\iconspacinglength}%
		}}}\ignorespaces%
	}%
	{%
	\end{mdframed}%
}



%
%
%	FORMATING SECTION
%
%
\renewcommand\chaptername{Practical}
\pagestyle{fancy}
\fancyhf{}
\fancyfoot[LE,RO]{\thepage}
\renewcommand{\footrulewidth}{1pt}
\fancyhead[LE]{\leftmark}
\fancyhead[RO]{\rightmark}

\geometry{
	top=1in,            % <-- you want to adjust this
	inner=1in,
	outer=3.5cm,
	bottom=3.5cm,
	headheight=3ex,       % <-- and this
	headsep=2ex,          % <-- and this
}

\raggedbottom %Remove it before printing as this is something to do with global settings. Can make each page look uneven but more dense. 
%\onehalfspacing
%\doublespacing
\makeindex

\title{Polygenic Risk Score Analyses Workshop}
\author{Dr Paul O'Reilly \and Dr Shing Wan Choi}


\begin{document}\thispagestyle{empty}
	\pagestyle{empty}
	\begin{titlepage}
		\begingroup% 
		\vfill
		\hbox{%
			\rule{1pt}{\dimexpr\textheight-28pt\relax}%
			\hspace*{0.1\textwidth}% 
			\parbox[b]{0.75\textwidth}{
				\vbox{
					\vspace{0.1\textheight}
					\begin{center}
						{\Huge\bfseries Polygenic Risk Score Analyses Workshop 2022\par}
						\vskip2.1\baselineskip
						\includegraphics[width=0.5\textwidth]{img/unim.png}\par
						\vspace{15mm}
						\Huge\bfseries Day 1: Introduction to PLINK
						\vspace{0.3\textheight}
					\end{center}
				}
			}
		}% end of hbox
		\vfill
		\null
		\endgroup
		
	\end{titlepage}
	%\frontmatter 
	
	\mainmatter
	\pagestyle{fancy}
	\setlength{\parindent}{4em}
	\setlength{\parskip}{0.75em}
		
	\chapter*{Day 1 Timetable}
	\label{chapter:info}
	\addcontentsline{toc}{chapter}{\nameref{chapter:info}}
	
	%\section*{Timetable}
	\label{sec:timetable}
	\addcontentsline{toc}{section}{\nameref{sec:timetable}}
	\begin{table}[h]
		\begin{tabular}{P{0.2\textwidth} P{0.5\textwidth}P{0.3\textwidth}}
			\toprule
			\textbf{Time} & \textbf{Title} & \textbf{Presenter} \\
			\midrule
			\vspace{9mm}
			9:00 - 9:15 & \begin{tabular}{l} Welcome Address \end{tabular} & Dr Daneshwar and Dr Baichoo   \\
			\vspace{11mm}
			9:15 - 9:30 & Opening Speech from Organisers & Dr Segun Fatumo and Dr Nicki Tiffin \\
			\vspace{9mm}
			9:30 - 10:30 & \underline{Lecture}: Background to PRS: GWAS \& relevant Statistics & Dr Paul O'Reilly \\
			\vspace{7mm}
			10:30 - 11:00 & Coffee Break and Q\&A & - \\
			\vspace{6mm}
			11:00 - 12:00 & \underline{Practical}: Introduction to Bash and R & Dr Paul O'Reilly \& Tutors \\
			\vspace{6mm}
			12:00 - 13:30 & Lunch & - \\
			\vspace{10mm}
			13:30 - 15:00 & \underline{Practical}: Introduction to PLINK I - Basics & Dr Conrad Iyegbe \& Tutors \\
			\vspace{6mm}
			15:00 - 15:30 & Coffee Break and Q\&A & - \\
			\vspace{7mm}
			15:30 - 16:30 & \underline{Practical}: Introduction to PLINK II - QC \& GWAS & Dr Conrad Iyegbe \& Tutors \\
			\bottomrule
		\end{tabular}
	\end{table}
	\tableofcontents
	\glsresetall	
	
	\chapter{Introduction to PLINK I: Basics}
	
	\section{Key Learning Outcomes}
	After completing this practical, you should be able to:
	\begin{enumerate}
		\item Explore and generate genetic data sets needed for GWAS
		\item Recode and reorder allelic data
		\item Use the PLINK website
		\item Select and exclude lists of samples and SNPs 
	\end{enumerate}
\vspace{3mm}
	\begin{warning}
		All data used in this workshop are \textbf{simulated}. 
		They have no specific biological meaning and are for demonstration purposes only. 
	\end{warning}
	
	\section{Introduction}
PLINK is the most popular software program for performing genome-wide association analyses Ð it is extremely extensive, allowing a huge number of analyses to be performed. It also includes many options for reformatting your data and provides useful data summaries. Software packages are usually best learnt by having a go at running some of their basic applications and progressing from there (rather than reading the entire user manual first!) - so we begin by running some basic PLINK commands and then work steadily towards performing more sophisticated analyses through these PLINK tutorials.
		
	\section{Command line basics}
In all of the instructions below, \emph{italics} indicate commands - they can be directly copy and pasted.  Anything in between the symbols $<>$ needs to be changed in some way.  For example, <file\_name> indicates that you should replace that entire statement (including the $<>$ symbols) with the appropriate file name.  \textbf{Bold} indicates non-command-line instructions (e.g. \textbf{right-click})

1.	Open up a terminal \\
2.	Navigate to the Practical/ folder and then the Day 1 working directory\\ (cd <directory\_name>) \\
3.	List all files in this directory by typing "ls" \\
4.	Test PLINK with no input by typing ./Software/plink \\
5.	Note that you can see lots of PLINK options by using the built-in help function: 
\begin{lstlisting}[language=bash]
./Software/plink --help 
\end{lstlisting}

\vspace{3mm}

\begin{note}
¥	Calling PLINK with no output will test if PLINK is installed and available in directory, because you should see some output showing the PLINK license and some commands. If you do not see this then please ask for help now!
\end{note}

\section{Exploring Data Sets}

1.	Open an Explorer window (`Finder' on a Mac) and navigate to your PLINK working directory. \\

\begin{note}
An explorer window should show same files as the `ls' command
\end{note}

\noindent 
2.	Open the file called `D1D.map' with a Text Editor e.g. by typing \textbf{right-click > Open}. \\
3.	Open the file `D1D.ped'.  Note this is a large file - if it will not open or is very slow, skip this step. \\
4.	Go to the PLINK website and investigate the format of the MAP/PED files (http://zzz.bwh.harvard.edu/plink/download.shtml) \\
(Look in the blue column on the left side)\\

\begin{questions}
What are the 4 columns in the map file?

What are the first 6 columns of ped file?

What information is in the remaining columns of the ped file?

\end{questions}
\noindent 
5.	Create `binary' format PLINK files using the recode command: \\
\begin{lstlisting}[language=bash]
./Software/plink
--file Data/D1D 
--make-bed 
--out Data/D1D
\end{lstlisting}

\noindent 
6.	List files (ls) and check which new files have appeared \\
7.	Open and examine files ending .bim and .fam. Do not open the .bed file.  \\
8.	Open and skim the `.log' file. \\

\begin{questions}
How is the fam file similar to the ped file?  How is it different?  Use the PLINK website if necessary.

How is the bim file similar to the map file?  How is it different?

\end{questions}

\section{Recoding alleles as counts}
Genotype data in allele count format is very useful, for example to use in regression modelling in statistical software such as R.\\[3mm]
\noindent 
1.	Generate the D1D data in allele count format:
\begin{lstlisting}[language=bash]
./Software/plink 
--bfile Data/D1D 
--recodeA 
--out Data/D1D_AC
\end{lstlisting}

\begin{note}
\noindent 
There are several options for recoding SNPs in different ways - more information on the PLINK website (see next section). \\

\noindent 
Again note that a log file was generated - skim the log file or screen output.

\end{note}

\begin{questions}
Look inside the .raw file. What do you think the 0/1/2Õs represent? 

Do there appear to be more 0s or more 2s? Why might this be?

\end{questions}

\section{PLINK website}

Go to http://zzz.bwh.harvard.edu/plink/download.shtml and skim through the front page to get an idea of PLINK's functionality. Note the list of clickable links on the left side of the website.

\noindent Under `Data Management' (click the heading on the left) and read the list of the different ways you may want to recode and reorder data sets.  Don't attempt to read much further as this is a very large and detailed section - a useful future resource but too much for today.

\noindent Under `Data Management', click `Write SNP list' and read the instructions there to write SNP lists.

\section{Write SNP list and extract SNPs}

You will now use the information that you found on the PLINK website to create a command to extract a list of SNPs.  Below is a list of requirements - try to do this before you go to the end of this section, where the full command is given and explained. \\

\noindent 
1.	Set the D1D binary file as input \\
2.	Set MAF threshold to 0.05 \\
3.	Set SNP missingness threshold to 0.05 \\
4.	Add the appropriate command to write out a snp list containing only those SNPs with MAF above 0.05 and missingness below 0.05 \\
5.	Use `D1D\_snps' as the output file name \\
6.	After the command has run, check the output for your SNP list and look at it with the default viewer. \\

\noindent You will now use the SNP list that you have created to extract those SNPs and create a new set of data files in a single command. \\

\noindent 
1.	Use the D1D binary file set as input \\
2.	Find the command for extracting a set of SNPs listed in a file (hint: Data Management section) and combine it with a command that you learned above to create binary files \\
3.	Use the output file name `D1D\_MAF\_MISS' \\

\begin{note}
\noindent 
Log files are useful to check that the number of SNPs and samples is as expected.  Always check your log files to be sure that they are sensible. 

\noindent 
SNP lists can also be used to EXCLUDE SNPs - select `exclude' above instead of `extract'. 

\noindent 
Sample ID lists can also be used to `keep' or `remove' individuals in the same `Filter' window.  Note that BOTH sample IDs (FID & IID, separated by a space) are required in the sample list file.

\end{note}
\noindent 
Solutions: \\
\begin{lstlisting}[language=bash]
./Software/plink
--bfile Data/D1D 
--maf 0.05 
--geno 0.05 
--write-snplist 
--out Data/D1D_snps
\end{lstlisting}
\vspace{2mm}

\begin{lstlisting}[language=bash]
./Software/plink 
--bfile Data/D1D 
--extract Data/D1D_snps.snplist 
--make-bed 
--out Data/D1D_MAF_MISS
\end{lstlisting}

\chapter{Introduction to PLINK II: Performing QC \& GWAS}

	\section{Key Learning Outcomes}
	After completing this practical, you should be able to:
	\begin{enumerate}
		\item Generate summaries of the data needed for QC
		\item Apply QC thresholds 
		\item Perform GWAS 
	\end{enumerate}

\section{Generate summaries to perform QC}
There are many kinds of summaries of the data that can generated in PLINK in order to perform particular quality control (QC) steps, which help to make our data more reliable. Some of these involve summaries in relation to the individuals (e.g. individual missingness, sex-check) and some relate to summaries of SNP data (e.g. MAF, Hardy-Weinburg Equilibrium). Over the next few sub-sections you will go through some examples of generating summary statistics that can be used to perform QC.

\subsection{Individual missingness}
\noindent
1.	Use the D1D binary files to generate files containing missingness information ($--$missing). Use the output file name `D1D\_miss' \\
\noindent
2.	Open the 2 files that were generated (lmiss \& imiss). \\

\begin{questions}
What do the two output files contain? 

In the imiss file, what is the meaning of the data in the column headed "F\_MISS"? 

\end{questions}

\vspace{5mm}

\subsection{SNP Missingness}
\noindent 
1.	Use the D1D binary files to generate files containing missingness information ($--$missing). Use the output file name `D1D\_miss' \\
\noindent 
2.	Look inside the file containing SNP missingness information: D1D\_miss.lmiss. \\

\begin{questions}
What is the meaning of the value under the heading F\_MISS? 

What does the command `$--$test$-$missing' do, and why might it be useful?

\end{questions}

\subsection{Hardy-Weinberg Equilibrium}
\noindent 
1.	Generate HWE statistics using the $--$hardy option. Use output file name D1D\_hardy.\\
\noindent 
2.	Open and examine results. \\

\begin{questions}
Why are there multiple rows for each SNP, and what does each mean?

Which of the rows do you think should be used to exclude SNPs from the subsequent association analysis (if any) for failing the HWE test?  Why?

\end{questions}

\subsection{Allele frequencies}
\noindent 
1.	Generate allele frequencies using the command $--$freq.  Use D1D\_freq as the output name.\\
\noindent
2.	Examine the output.

\begin{questions}
What is the heading of the column that tells you which nucleotide is the minor allele?

*Note - this is important to remember as many PLINK files use this notation - the minor allele is always labelled the same way

\end{questions}

\section{Apply QC filters} 
\textbf{There are different strategies for performing QC on your data:} \\[3mm]
\noindent 
(a)	create lists of SNPs and individuals and use $--$remove, $--$extract, $--$exclude, $--$include to create new file sets (good for documentation, collaboration) \\[3mm]
\noindent 
(b)	apply thresholds one at a time and generate new bed/bim/fam files (good for applying sequential filters) \\[3mm]
\noindent 
(c)	use options (e.g. $--$maf ) in other commands (e.g. $--$assoc) to remove SNPs or samples at required QC thresholds during analysis.\\

\begin{note}
We have already seen how to select or exclude individuals or SNPs by first creating lists (a), so in this section we will set thresholds to generate new file sets in a single command.  However, it is useful to have lists of all SNPs and individuals excluded pre-analysis, according to the reason for exclusion, so generating and retaining such files using the techniques that we used before is good practice. 
\end{note}

\subsection{Apply individual missingness thresholds}
1.	Generate new binary file sets ($--$make$-$bed) from the `D1D' binary file set, removing individuals with missingness greater than 3\% using a single command (hint: In the `Inclusion thresholds' section, see the `Missing/person' sub-section).  Use the output file name `D1D\_imiss3pc' \\
2.	Examine the output files (no need to open, and remember the bed file can not be read) and the log file

\begin{questions}
How many individuals were in the original file? \\

How many individuals were removed? \\

How many males and females were left after screening? 

\end{questions}

\subsection{Apply SNP missingness and MAF thresholds}
1.	Create new binary file sets from the `D1D\_imiss3pc' binary file set (NOT the original D1D files) by setting MAF threshold to 0.05 and SNP missingness threshold to 0.02 (See `Inclusion thresholds' to obtain the correct threshold flags). 
Use the output file name `D1D\_imiss3pc\_lmiss2pc\_maf5pc \\
2.	Examine the output files and the log file \\

\begin{questions}
How many SNPs were in the original file? 

How many SNPs were removed for low minor allele frequency?

How many SNPs were removed for missingness?

\end{questions}

\subsection{Apply Hardy-Weinberg thresholds}
1.	Generate a new binary file set called `D1D\_QC' from the D1D\_imiss3pc\_lmiss2pc\_maf5pc file, applying a HWE threshold of 0.0001.\\
2.	This is our final, QC'ed file set.\\
3.	Examine log and output files.

\begin{questions}
How many SNPs were removed for HWE $P$-values below the threshold?

NOTE - it is useful to know how to do this but be careful about setting this threshold - strong association signals can cause departures from HWE and you may remove great results!  Use a lenient threshold and apply to controls only to avoid this problem.  HWE can also be checked post-hoc for each SNP.
\end{questions}

\section{Perform GWAS} 

\subsection{Case/Control GWAS - no covariates}
Run the following code, which performs a genetic association study using logistic regression on some case/control data: \\

\begin{lstlisting}[language=bash]
./Software/plink 
--plink 
--bfile D1D_QC  	
--logistic 
--adjust 
--pheno D1D.pheno1 
--out Results/D1D_CC
\end{lstlisting}

\begin{questions}
What are the raw and Bonferonni-adjusted P-values for the top hit?   What does this mean - is there a significant association? 

Are there any other significant associations?

\end{questions}

\subsection{Case/Control GWAS - with covariates}
Here we repeat the previous analysis but this time including some covariates.  The file D1D.pcs1234 contains the first 4 principal components from a PCA on the genetic data. \\

\noindent 1.	Run the analysis specifying the covariates file:

\begin{lstlisting}[language=bash]
./Software/plink 
--plink 
--bfile D1D_QC  
--logistic 
--adjust 
--pheno D1D.pheno1 
--covar D1D.pcs.1234 
--out Results/D1D_CC_PCadj
\end{lstlisting}

\begin{questions}
What are the raw and Bonferonni-adjusted P-values for the top hit?   What does this mean - is there a significant association? 

Suggest a reason for the different results when adjusting for the 4 PCs.

\end{questions}


	\chapter*{License}
	\label{chapter:license}
	\addcontentsline{toc}{chapter}{\nameref{chapter:license}}
	This work is licensed under Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International Public License and the below text is a summary of the main terms of the full Legal Code (the full licence) available at \url{https://creativecommons.org/licenses/by-nc-sa/4.0/legalcode}.
	
	\noindent\textbf{You are free to:}
	\begin{itemize}
		\item \textbf{Share} â€” copy and redistribute the material in any medium or format
		\item \textbf{Adapt} â€” remix, transform, and build upon the material
	\end{itemize}
	
	\noindent The licensor cannot revoke these freedoms as long as you follow the license terms.
	
	\noindent\textbf{Under the following terms:}
	\begin{itemize}
		\item \textbf{Attribution} â€” You must give appropriate credit, provide a link to the license, and indicate if changes were made. You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.
		\item	\textbf{NonCommercial} â€” You may not use the material for commercial purposes.
		\item \textbf{ShareAlike} â€” If you remix, transform, or build upon the material, you must distribute your contributions under the same license as the original.
		
	\end{itemize}
	
	\noindent No additional restrictions â€” You may not apply legal terms or technological measures that legally restrict others from doing anything the license permits.
	
	
	\noindent\textbf{Notices:}
	
	\noindent You do not have to comply with the license for elements of the material in the public domain or where your use is permitted by an applicable exception or limitation.
	
	\noindent No warranties are given. The license may not give you all of the permissions necessary for your intended use. For example, other rights such as publicity, privacy, or moral rights may limit how you use the material.

	\backmatter
	\printbibliography[heading=bibintoc,title={Reference}]
	%\printglossary[title=Index]
\end{document}
