> **Polygenic Risk Score Analyses Workshop 2022**

![](media/image1.jpeg){width="2.939998906386702in"
height="2.8874989063867016in"}

> **Day 2: Introduction to Polygenic Risk Scores**

# Day 2 Timetable {#day-2-timetable .unnumbered}

+------------+--------+-------------------------+---------------------+
| > **Time** | > **T  |                         | > **Presenter**     |
|            | itle** |                         |                     |
+============+========+=========================+=====================+
| > 9:00 -   | A      |                         | > \-                |
| > 9:15     | rrival |                         |                     |
+------------+--------+-------------------------+---------------------+
| > 9:15 -   | [Lec   | > Introduction to PRS I | > Dr Paul O'Reilly  |
| > 10:30    | ture]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
+------------+--------+-------------------------+---------------------+
| 10:30 -    | >      |                         | > \-                |
| 11:00      | Coffee |                         |                     |
|            | >      |                         |                     |
|            |  Break |                         |                     |
|            | > and  |                         |                     |
|            | > Q&A  |                         |                     |
+------------+--------+-------------------------+---------------------+
| 11:00 -    | > [Lec |                         | > Dr Paul O'Reilly  |
| 12:00      | ture]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | >      |                         |                     |
|            | Introd |                         |                     |
|            | uction |                         |                     |
|            | > to   |                         |                     |
|            | > PRS  |                         |                     |
|            | > II   |                         |                     |
+------------+--------+-------------------------+---------------------+
| 12:00 -    | >      |                         | > \-                |
| 13:30      |  Lunch |                         |                     |
+------------+--------+-------------------------+---------------------+
| 13:30 -    | >      |                         | > Dr Conrad Iyegbe  |
| 14:30      | [Pract |                         | > & Tutors          |
|            | ical]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | >      |                         |                     |
|            | Introd |                         |                     |
|            | uction |                         |                     |
|            | > to   |                         |                     |
|            | > PRS  |                         |                     |
|            | > I    |                         |                     |
+------------+--------+-------------------------+---------------------+
| 14:30 -    | >      |                         | > \-                |
| 15:00      | Coffee |                         |                     |
|            | >      |                         |                     |
|            |  Break |                         |                     |
|            | > and  |                         |                     |
|            | > Q&A  |                         |                     |
+------------+--------+-------------------------+---------------------+
| 15:00 -    | >      |                         | > Dr Conrad Iyegbe  |
| 16:00      | [Pract |                         | > & Tutors          |
|            | ical]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | >      |                         |                     |
|            | Introd |                         |                     |
|            | uction |                         |                     |
|            | > to   |                         |                     |
|            | > PRS  |                         |                     |
|            | > II   |                         |                     |
+------------+--------+-------------------------+---------------------+
| 16:00 -    | > [S   |                         | > Dr Nicki Tiffin   |
| 17:00      | pecial |                         |                     |
|            | > Sem  |                         |                     |
|            | inar]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | > Pol  |                         |                     |
|            | ygenic |                         |                     |
|            | > Risk |                         |                     |
|            | >      |                         |                     |
|            | Scores |                         |                     |
|            | >      |                         |                     |
|            | > \+   |                         |                     |
|            | > and  |                         |                     |
|            | >      |                         |                     |
|            | Ethics |                         |                     |
+------------+--------+-------------------------+---------------------+

> **Contents**

[Day 2 Timetable](#day-2-timetable) 1

[Day 2 Timetable](#day-2-timetable) . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . 1

1.  [Introduction to Polygenic Score
    > Analyses](#introduction-to-polygenic-score-analyses) 3

    1.  [Key Learning Outcomes](#key-learning-outcomes) . . . . . . . .
        . . . . . . . . . . . . . . . . 3

    2.  [Resources you will be using](#resources-you-will-be-using) . .
        . . . . . . . . . . . . . . . . . . . . 3

    3.  [Data Structure](#data-structure) . . . . . . . . . . . . . . .
        . . . . . . . . . . . . . . 4

    4.  [Introduction](#introduction) . . . . . . . . . . . . . . . . .
        . . . . . . . . . . . . . . 5

    5.  [Understanding GWAS Summary
        Statistics](#understanding-gwas-summary-statistics) . . . . . .
        . . . . . . . . 5

    6.  [Matching the Base and Target Data
        sets](#matching-the-base-and-target-data-sets) . . . . . . . . .
        . . . . . . 6

    7.  [Linkage Disequilibrium in PRS
        Analyses](#linkage-disequilibrium-in-prs-analyses) . . . . . . .
        . . . . . . . . 7

        1.  [Performing Clumping](#performing-clumping) . . . . . . . .
            . . . . . . . . . . . . . . 7

    8.  [P-Value Thresholding](#p-value-thresholding) . . . . . . . . .
        . . . . . . . . . . . . . . . . . 8

        1.  [Height PRS using GW-significant SNPs
            only](#height-prs-using-gw-significant-snps-only) . . . . .
            . . . . 8

        2.  [Height PRS across multiple P-value
            thresholds](#height-prs-across-multiple-p-value-thresholds)
            10

        3.  [High Resolution Scoring](#high-resolution-scoring) 12

    9.  [Stratifying Samples by PRS](#stratifying-samples-by-prs) 15

    10. [Case Control Studies](#case-control-studies) 17

    11. [Cross-Trait Analysis](#cross-trait-analysis) 18

# Introduction to Polygenic Score Analyses

## Key Learning Outcomes

> After completing this practical, you should be able to:

1.  Perform basic Polygenic Risk Score (PRS) analyses using PRSice
    (Euesden, Lewis & O'Reilly 2015; Choi & O'Reilly 2019)

2.  Interpret results generated from PRS analyses

3.  Customise visualisation of results

## Resources you will be using

> To perform PRS analyses, summary statistics from Genome Wide
> Association Stud- ies (GWAS) are required. In this workshop, the
> following summary statistics are used:
>
> Phenotype Provider Description Download Link
>
> Height [GIANT
> Consortium](https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files)
> GWAS of height on
>
> 253,288 individuals (**wood_defining_2014**)
>
> [Download](https://portals.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz)
> [Link](https://portals.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz)
>
> Coronary artery disease (CAD)
>
> [CARDIoGRAM plus](http://www.cardiogramplusc4d.org/) [C4D
> consortium](http://www.cardiogramplusc4d.org/)
>
> GWAS on 60,801
>
> CAD cases and 123,504 controls
>
> [Download](http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip)
> [Link](http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip)

#### (consortium_comprehensive_2015) {#consortium_comprehensive_2015 .unnumbered}

## Data Structure

> You will find all practical materials in the **PRS_Workshop/Day_2**
> directory. Relevant materials that you should see there at the start
> of the practical are as follows:
>
> ![](media/image2.png)Practical Base_Data
>
> GIANT_Height.txt cad.add.txt cad.add.readme
>
> Target_Data
>
> TAR.fam TAR.bim TAR.bed TAR.height
>
> TAR.cad TAR.covariate
>
> Software
>
> plink_mac plink_linux plink.exe
>
> PRSice.R PRSice_mac PRSice_linux PRSice_win64.exe

![](media/image5.jpeg){width="0.3229166666666667in"
height="0.3229166666666667in"}

## Introduction

> A PRS is a (usually weak) estimate of an individual's genetic
> propensity to a pheno- type, calculated as a sum of their genome-wide
> genotypes weighted by corresponding genotype effect sizes obtained
> from GWAS summary statistics. In the next section we will consider
> what the effect size means and how it is used in computing PRS.

## Understanding GWAS Summary Statistics

> When GWAS are performed on a quantitative trait, the effect size is
> typically given as a beta coefficient (𝛽) from a linear regression
> with Single Nucleotide Polymor- phism (SNP) genotypes as predictor of
> phenotype. The 𝛽 coefficient estimates the increase in the phenotype
> for each copy of the *effect allele*. For example, if the effect
> allele of a SNP is **G** and the non-effect allele is **A** , then the
> genotypes **AA**, **AG** and **GG** will be coded as 0, 1 and 2
> respectively. In this scenario, the 𝛽 coefficient reflects how much
> the phenotype changes for each **G** allele present (NB. The 𝛽 can be
> positive or negative - so the 'effect allele' is simply the allele
> that was coded in the regression, not necessarily the allele with a
> positive effect).
>
> When a GWAS is performed on a binary trait (e.g. case-control study),
> the effect size is usually reported as an Odd Ratios (OR). Using the
> same example, if the OR from the GWAS is 2 with respect to the **G**
> allele, then the OR of **AG** relative to **AA** is 2, and the OR of
> **GG** relative to **AA** is 4. So an individual with the **GG**
> genotype are estimated\* to be 4 times more likely to be a case than
> someone with the **AA** genotype (\*an Odds Ratio is itself an
> estimate of a Risk Ratio, which cannot be calculated from a
> case/control study)

![](media/image6.jpeg){width="0.3281244531933508in"
height="0.3281244531933508in"}

> ![](media/image7.jpeg){width="0.3541655730533683in"
> height="0.3593744531933508in"}

## Matching the Base and Target Data sets

> The first step in PRS calculation is to ensure consistency between the
> GWAS sum- mary statistic file (*base data*) and the target genotype
> file (*target data*). Since the base and target data are generated
> independently, they often relate to different SNPs
>
> \- and so the first job is to identify the overlapping SNPs across the
> two data sets and remove non-overlapping SNPs (this is usually done
> for you by PRS software). If the overlap is low then it would be a
> good idea to perform imputation on your target data to increase the
> number of SNPs that overlap between the data sets.
>
> The next, more tricky issue, is that the genotype encoding between the
> data sets may differ. For example, while the effect allele of a SNP is
> **T** in the base data, the effect allele in the target might be **G**
> instead. When this occurs, *allele flipping* should be performed,
> where the genotype encoding in the target data is reversed so that
> **TT**, **TG** and **GG** are coded as 2, 1 and 0. Again, this is
> usually performed automatically by PRS software.

![](media/image5.jpeg){width="0.3229166666666667in"
height="0.3229166666666667in"}

## Linkage Disequilibrium in PRS Analyses

> GWAS are typically performed one-SNP-at-a-time, which, combined with
> the strong correlation structure across the genome (Linkage
> Disequilibrium (LD)), makes iden- tifying the independent genetic
> effects (or their best proxies if these are not geno- typed/imputed)
> challenging. There are two main options for approximating the PRS that
> would have been generated from full conditional GWAS: 1. SNPs are
> *clumped* so that the retained SNPs are largely independent of each
> other, allowing their effects to be summed, assuming additive effects,
> 2. all SNPs are included and the LD between them is accounted for.
>
> While option 2 is statistically appealing, option 1 has been most
> adopted in PRS studies so far, most likely due to its simplicity and
> the similarity of results of meth- ods using the different options to
> date (**mak_polygenic_2017**). In this work- shop we will consider
> option 1, implemented in PRSice, but if you are interested in how LD
> can be incorporated as a parameter in PRS calculation then see the LD-
> pred (**vilhjalmsson_modeling_2015**) and lassosum
> (**mak_polygenic_2017**) papers.

### Performing Clumping

> *Clumping* is the procedure where a SNP data set is 'thinned' by
> removing SNPs across the genome that are correlated (in high LD) with
> a nearby SNP that has a smaller association 𝑃 -value.
>
> SNPs are first sorted (i.e. ranked) by their 𝑃 -values. Then, starting
> from the most significant SNP (denoted as the *index SNP*), any SNPs
> in high LD (eg. 𝑟^2^ \> 0.1, with 𝑟^2^ typically calculated from
> 𝑝ℎ𝑎𝑠𝑒𝑑 ℎ𝑎𝑝𝑙𝑜𝑡𝑦𝑝𝑒 data) with the index SNP are removed. To reduce
> computational burden, only SNPs that are within e.g. 250 kb of the
> *index SNP* are 𝑐𝑙𝑢𝑚𝑝𝑒𝑑. This process is continued until no *index
> SNPs* remain.
>
> Use the command below to perform clumping of the Height GWAS data
> using PLINK(**chang_second_2015**). First, you will have to navigate
> to the right folder where the data are stored using the terminal. Open
> the terminal and type the command below at the terminal prompt:
>
> 1 cd \~/Desktop/PRS\\\_Workshop/
>
> Next type the following command (NB. See warning below):
>
> 1 ./Software/plink_linux
>
> 2 \--bfile Target_Data/TAR
>
> 3 \--clump Base_Data/GIANT_Height.txt
>
> 4 \--clump-p1 1
>
> 5 \--clump-snp-field MarkerName
>
> 6 \--clump-field p
>
> 7 \--clump-kb 250
>
> 8 \--clump-r2 0.1
>
> 9 \--out Results/Height

![](media/image5.jpeg){width="0.3229155730533683in"
height="0.3229166666666667in"}

> The command above performs clumping on the height GWAS using LD
> calculated based on the **TAR** genotype file. SNPs that have 𝑟^2^ \>
> 0.1 within a 250 kb window of the index SNP are removed. This will
> generate the **Height.clumped** file, which contains the SNPs retained
> after clumping.

![](media/image7.jpeg){width="0.3541655730533683in"
height="0.3593744531933508in"}

## P-Value Thresholding

> Deciding which SNPs to include in the calculation of PRS is one of the
> major challenges in the field. A simple and popular approach is to
> include SNPs according to their GWAS association 𝑃 -value. For
> example, we may choose to include only the genome-wide significant
> SNPs from the GWAS because those are the SNPs with significant
> evidence for association. In the next subsection you will compute PRS
> from GW-significant SNPs only, and then in the subsequent subsection
> you will generate multiple PRSs using different 𝑃 -value thresholds.

### Height PRS using GW-significant SNPs only

> Use the commands below to run PRSice with GIANT Height GWAS as base
> data and the height phenotype target data. PRSice will calculate
> Height PRS in the
>
> target data and then perform a regression of the Height PRS against
> the target individual's true height values. From the
> **PRS_Workshop/Day_2** directory, run the following command in the
> terminal:
>
> 1 Rscript ./Software/PRSice.R
>
> 2 \--prsice Software/PRSice_linux
>
> 3 \--base Base_Data/GIANT_Height.txt
>
> 4 \--target Target_Data/TAR
>
> 5 \--snp MarkerName
>
> 6 \--A1 Allele1
>
> 7 \--A2 Allele2
>
> 8 \--stat b
>
> 9 \--beta
>
> 10 \--pvalue p
>
> 11 \--pheno Target_Data/TAR.height
>
> 12 \--binary-target F
>
> 13 \--bar-levels 5e-8
>
> 14 \--no-full
>
> 15 \--fastscore
>
> 16 \--out Results/Height.gws
>
> This command takes the Height GWAS summary statistic file (\--base),
> informs PRSice of the column name for the column containing the SNP ID
> (\--snp), the effect allele (\--A1), the non-effect allele (\--A2),
> the effect size (\--stat) and the 𝑃 -value (\--pvalue). We also inform
> PRSice that the effect size is a 𝛽 coefficient (\--beta) instead of an
> OR. The \--binary-target F command informs PRSice that the target
> phenotype is a quantitative trait and thus linear regression should be
> performed. In addi- tion, we ask PRSice not to perform high-resolution
> scoring over multiple thresholds
>
> (\--fastscore), and to compute the PRS using only those SNPs with 𝑃
> -value \< 5*×*10^*−*8^.

![](media/image6.jpeg){width="0.3281244531933508in"
height="0.3281244531933508in"}

> PRSice performs strand flipping and clumping automatically and
> generates the **Height.gws.summary** file, together with other output
> that we will look into later in the practical. The summary file
> contains the following columns:

1.  **Phenotype** - Name of Phenotype.

2.  **Set** - Name of Gene Set. Default is *Base*

3.  **Threshold** - Best P-value Threshold

4.  **PRS.R2** - Variance explained by the PRS

5.  **Full.R2** - Variance explained by the full model (including the
    covariates)

6.  **Null.R2** - Variance explained by the covariates (none provided
    here)

7.  **Prevalence** - The population disease prevalence as indicated by
    the user (not provided here due to testing continuous trait)

8.  **Coefficient** - The 𝛽 coefficient corresponding to the effect
    estimate of the best-fit PRS on the target trait in the regression.
    A one unit increase in the PRS increases the outcome by 𝛽

9.  **Standard.Error** - The standard error of the best-fit PRS 𝛽
    coefficient (see above)

10. **P** - The 𝑃 -value relating to testing the null hypothesis that
    the best-fit PRS

> 𝛽 coefficient is zero.

11. **Num_SNP** - Number of SNPs included in the best-fit PRS

12. **Empirical-P** - Only provided if permutation is performed. This is
    the em- pirical 𝑃 -value corresponding to the association test of
    the best-fit PRS - this controls for the over-fitting that occurs
    when multiple thresholds are tested.

> For now, we can ignore most columns and focus on the **PRS.R2** and
> the **P** column, which provide information on the model fit.

![](media/image7.jpeg){width="0.3541666666666667in"
height="0.3593744531933508in"}

### Height PRS across multiple P-value thresholds

> A disadvantage of using only genome-wide significant SNPs is that
> there are likely to be many true signals among SNPs that did not reach
> genome-wide significance. However, since we do not know what 𝑃 -value
> threshold provides the \"best\" predic- tion for our particular data,
> then we can calculate the PRS under several 𝑃 -value thresholds and
> test their prediction accuracy to identify the \"best\" threshold (NB.
>
> []{#_bookmark13 .anchor}Figure 1.1: BARPLOT generated by PRSice

![](media/image8.png){width="2.7440616797900264in"
height="2.761874453193351in"}

> See **dudbridge_power_2013** for theory on factors affecting the
> best-fit PRS). This process is implemented in PRSice and can be
> performed automatically as fol- lows:
>
> 1 Rscript ./Software/PRSice.R
>
> 2 \--dir .
>
> 3 \--prsice Software/PRSice_linux
>
> 4 \--base Base_Data/GIANT_Height.txt
>
> 5 \--target Target_Data/TAR
>
> 6 \--snp MarkerName
>
> 7 \--A1 Allele1
>
> 8 \--A2 Allele2
>
> 9 \--stat b
>
> 10 \--beta
>
> 11 \--pvalue p
>
> 12 \--pheno Target_Data/TAR.height
>
> 13 \--binary-target F
>
> 14 \--fastscore
>
> 15 \--out Results/Height.fast
>
> By removing the \--bar-levels and \--no-full command, we ask PRSice to
> perform PRS calculation with a number of predefined thresholds (0.001,
> 0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 1). The **.prsice** file is very
> similar to the **.summary** file, the only difference is that
>
> **.prsice** file reports the results of model fits for **all
> thresholds** instead of the most predictive threshold. This allow us
> to observe the change in model fitting across different thresholds,
> visualized by the BARPLOT (fig. [1.1)](#_bookmark13) generated by
> PRSice
>
> ![](media/image7.jpeg){width="0.3541655730533683in"
> height="0.3593744531933508in"}

### High Resolution Scoring

> If we limit ourselves to a small number of 𝑃 -value thresholds, we
> might \"miss\" the most predictive threshold. In order to identify
> this \"best\" threshold, we will need \"high-resolution scoring\",
> that is, to test the predictive power of PRS generated under a large
> number of p-value thresholds. We can achieve that by simply removing
> the \--fastscore command from the PRSice script:
>
> 1 Rscript ./Software/PRSice.R
>
> 2 \--dir .
>
> 3 \--prsice Software/PRSice_linux
>
> 4 \--base Base_Data/GIANT_Height.txt
>
> 5 \--target Target_Data/TAR
>
> 6 \--snp MarkerName
>
> 7 \--A1 Allele1
>
> 8 \--A2 Allele2
>
> 9 \--stat b
>
> 10 \--beta
>
> 11 \--pvalue p
>
> 12 \--pheno Target_Data/TAR.height
>
> 13 \--binary-target F
>
> 14 \--out Results/Height.highres
>
> When PRSice performs high-resolution scoring, it will generate a plot
> (fig. [1.2)](#_bookmark15) presenting the model fit of PRS calculated
> at all P-value thresholds.
>
> []{#_bookmark15 .anchor}Figure 1.2: High Resolution Plot generated by
> PRSice

![](media/image9.jpeg){width="2.463332239720035in"
height="2.455in"}![](media/image7.jpeg){width="0.3541655730533683in"
height="0.3593744531933508in"}

> **Accounting for Covariates**
>
> When performing PRS, one might want to account for covariates. Based
> on user in- puts, PRSice can automatically incorporate covariates into
> its model. For example, the following commands will include sex into
> the regression model as a covariate:
>
> 1 Rscript ./Software/PRSice.R
>
> 2 \--prsice Software/PRSice_linux
>
> 3 \--base Base_Data/GIANT_Height.txt
>
> 4 \--target Target_Data/TAR
>
> 5 \--snp MarkerName
>
> 6 \--A1 Allele1
>
> 7 \--A2 Allele2
>
> 8 \--stat b
>
> 9 \--beta
>
> 10 \--pvalue p
>
> 11 \--pheno Target_Data/TAR.height
>
> 12 \--binary-target F
>
> 13 \--cov Target_Data/TAR.covariate
>
> 14 \--cov-col Sex
>
> 15 \--out Results/Height.sex
>
> When covariates were include in the analysis, PRSice will use the
> model fit of **only PRS** for all its output. This 𝑃 𝑅𝑆.𝑅^2^ is
> calculated by minusing the
>
> 𝑅^2^ of the null model (e.g. 𝐻𝑒𝑖𝑔ℎ𝑡 *∼* 𝑆𝑒𝑥) from the 𝑅^2^ of the full
> model (e.g.
>
> ![](media/image10.jpeg){width="0.34781167979002625in"
> height="0.34781167979002625in"}![](media/image7.jpeg){width="0.3541666666666667in"
> height="0.3593744531933508in"}𝐻𝑒𝑖𝑔ℎ𝑡 *∼* 𝑆𝑒𝑥 + 𝑃 𝑅𝑆).
>
> Usually, with categorical variables, dummy variables have to be
> generated to rep- resent the different categories. Alternatively,
> residualized phenotype, generated by regresing the covariates against
> the phenotype, can be used for downstream anal- yses. A useful feature
> of PRSice is to automatically generate the dummy variable for users.
> This can be achieved with the following command:
>
> 1 Rscript.exe ./Software/PRSice.R
>
> 2 \--prsice Software/PRSice_linux
>
> 3 \--base Base_Data/GIANT_Height.txt
>
> 4 \--target Target_Data/TAR
>
> 5 \--snp MarkerName
>
> 6 \--A1 Allele1
>
> 7 \--A2 Allele2
>
> 8 \--stat b
>
> 9 \--beta
>
> 10 \--pvalue p
>
> 11 \--pheno Target_Data/TAR.height
>
> 12 \--binary-target F
>
> 13 \--cov Target_Data/TAR.covariate
>
> 14 \--cov-col Sex
>
> 15 \--cov-factor Sex
>
> 16 \--out Results/Height.sex
>
> []{#_bookmark16 .anchor}Figure 1.3: Example of a quantile plot
> generated by PRSice

![](media/image11.png){width="2.4641655730533683in"
height="2.4674989063867017in"}

## Stratifying Samples by PRS

> An interesting application of PRS is to test whether samples with
> higher PRS have higher phenotypic values. This can be nicely
> visualized using the quantile plot (fig. [1.3).](#_bookmark16)
>
> To generate quantile plots in PRSice, simply add \--quantile 10
> option.

![](media/image6.jpeg){width="0.3281244531933508in"
height="0.3281244531933508in"}

> 1 Rscript ./Software/PRSice.R
>
> 2 \--prsice Software/PRSice_linux
>
> 3 \--base Base_Data/GIANT_Height.txt
>
> 4 \--target Target_Data/TAR
>
> 5 \--snp MarkerName
>
> 6 \--A1 Allele1
>
> 7 \--A2 Allele2
>
> 8 \--stat b
>
> 9 \--beta
>
> 10 \--pvalue p
>
> 11 \--pheno Target_Data/TAR.height
>
> 12 \--binary-target F
>
> 13 \--cov Target_Data/TAR.covariate
>
> 14 \--cov-col Sex
>
> 15 \--plot
>
> 16 \--quantile 10
>
> 17 \--out Results/Height.sex

![](media/image10.jpeg){width="0.34781167979002625in"
height="0.34781167979002625in"}

> Figure 1.4: Example of a strata plot generated by PRSice

![](media/image12.jpeg){width="2.4641655730533683in"
height="2.4674989063867017in"}

> A disadvantage of the quantile plot is that it only seperate samples
> into quantiles of equal size. However, it is somtimes interesting to
> investigate whether a specific strata (e.g. top 5% of samples),
> contain a higher PRS than the reference strata. For example,
> **mavaddat_prediction_2015** found that samples in the highest 1% of
> PRS distribution have a 2.81 increased OR of breast cancer when
> comparing to samples at the middle quantiles (40th to 60th
> percentile). We can mimic their table by using \--quant-break, which
> represents the upper bound of each strata, and \--quant-ref, which
> represents the upper bound of the reference quantile:
>
> 1 Rscript ./Software/PRSice.R
>
> 2 \--prsice Software/PRSice_linux
>
> 3 \--base Base_Data/GIANT_Height.txt
>
> 4 \--target Target_Data/TAR
>
> 5 \--snp MarkerName
>
> 6 \--A1 Allele1
>
> 7 \--A2 Allele2
>
> 8 \--stat b
>
> 9 \--beta
>
> 10 \--pvalue p
>
> 11 \--pheno Target_Data/TAR.height
>
> 12 \--binary-target F
>
> 13 \--cov Target_Data/TAR.covariate
>
> 14 \--cov-col Sex
>
> 15 \--plot
>
> 16 \--quantile 100
>
> 17 \--quant-break 1,5,10,20,40,60,80,90,95,99,100
>
> 18 \--quant-ref 60
>
> 19 \--out Results/Height.sex

![](media/image6.jpeg){width="0.3281244531933508in"
height="0.3281244531933508in"}

## Case Control Studies

> In the previous exercises, we have performed PRS analyses on height,
> which is a quantitative trait. For binary phenotypes (e.g
> case-control) there are a number of differences in the analysis:

1.  Logistic regression has to be performed instead of linear regression

2.  ORs are usually provided and need to be converted to 𝛽's when
    constructing PRS

> Here we will use CAD as an example. You will find the summary
> statistic under *Base_Data* (**cad.add.txt**) and the phenotype file
> (**TAR.cad**) under *Tar- get_Data*. You will also need to specify
> \--binary-target T in the PRSice command to indicate that the
> phenotype is binary.

![](media/image10.jpeg){width="0.34781167979002625in"
height="0.34781167979002625in"}

> 1 Rscript ./Software/PRSice.R
>
> 2 \--prsice Software/PRSice_linux
>
> 3 \--base Base_Data/cad.add.txt
>
> 4 \--target Target_Data/TAR
>
> 5 \--snp markername
>
> 6 \--A1 effect_allele
>
> 7 \--A2 noneffect_allele
>
> 8 \--chr chr
>
> 9 \--bp bp_hg19
>
> 10 \--stat beta
>
> 11 \--beta
>
> 12 \--pvalue p_dgc
>
> 13 \--pheno Target_Data/CAD.pheno
>
> 14 \--binary-target T
>
> 15 \--out Results/CAD.highres

![](media/image10.jpeg){width="0.34781167979002625in"
height="0.34781167979002625in"}

> ![](media/image13.jpeg){width="3.5316666666666667in"
> height="2.066457786526684in"}
>
> []{#_bookmark19 .anchor}Figure 1.5: Plot taken from Ruderfer et al.
> 2014

![](media/image7.jpeg){width="0.3541655730533683in"
height="0.3593744531933508in"}

## Cross-Trait Analysis

> A popular application of PRS is in performing cross-trait analyses.
> This allows some interesting analyses such as those performed by
> **ruderfer_polygenic_2014** (fig. [1.5),](#_bookmark19) which used the
> bipolar PRS to predict into different clinical dimensions of
> schizophrenia.
>
> In this practical, we will perform cross-trait analyses between CAD
> and Height, using height as the base and CAD as the target.

![](media/image10.jpeg){width="0.34781167979002625in"
height="0.34781167979002625in"}

> 1 Rscript ./Software/PRSice.R
>
> 2 \--prsice Software/PRSice_linux
>
> 3 \--base Base_Data/GIANT_Height.txt
>
> 4 \--target Target_Data/TAR
>
> 5 \--snp MarkerName
>
> 6 \--A1 Allele1
>
> 7 \--A2 Allele2
>
> 8 \--stat b
>
> 9 \--beta
>
> 10 \--pvalue p
>
> 11 \--pheno Target_Data/CAD.pheno
>
> 12 \--binary-target T
>
> 13 \--out Results/Cross.highres

![](media/image7.jpeg){width="0.3541655730533683in"
height="0.3593744531933508in"}
