<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="author" content="Tade Souaiaia" />
      <link rel="shortcut icon" href="../../../../../img/favicon.ico" />
    <title>Day2.docx - SibArc</title>
    <link rel="stylesheet" href="../../../../../css/theme.css" />
    <link rel="stylesheet" href="../../../../../css/theme_extra.css" />
        <link href="../../../../../css/extra.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Day2.docx";
        var mkdocs_page_input_path = "tmp/prs_2023-main/images/Day2.docxfolder/Day2.docx.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../../../../js/html5shiv.min.js"></script>
    <![endif]--> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../../../../.." class="icon icon-home"> SibArc
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../../../..">Home</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Quick Start</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../../quik_install/">Installation and Data</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../../quik_demo/">Running SibArc</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../../quik_result/">Interpreting Results</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">About</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../../about_future/">Future Work</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Misc</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../workshop_practical_paul.docx/">Day 1 (AM)</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../workshop_practical_tade/">Day 1 (PM)</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../../workshop_practical_clive.docx/">Day 2 (AM)</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../..">SibArc</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../../../../.." class="icon icon-home" aria-label="Docs"></a></li>
      <li class="breadcrumb-item active">Day2.docx</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/tadesouaiaia/sibArc-website/edit/master/docs/tmp/prs_2023-main/images/Day2.docxfolder/Day2.docx.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <blockquote>
<p><strong>Polygenic Risk Score Analyses Workshop 2022</strong></p>
</blockquote>
<p><img alt="" src="media/image1.jpeg" />{width="2.939998906386702in"
height="2.8874989063867016in"}</p>
<blockquote>
<p><strong>Day 2: Introduction to Polygenic Risk Scores</strong></p>
</blockquote>
<h1 class="unnumbered" id="day-2-timetable">Day 2 Timetable</h1>
<p>+------------+--------+-------------------------+---------------------+
| &gt; <strong>Time</strong> | &gt; <strong>T  |                         | &gt; **Presenter</strong>     |
|            | itle** |                         |                     |
+============+========+=========================+=====================+
| &gt; 9:00 -   | A      |                         | &gt; -                |
| &gt; 9:15     | rrival |                         |                     |
+------------+--------+-------------------------+---------------------+
| &gt; 9:15 -   | [Lec   | &gt; Introduction to PRS I | &gt; Dr Paul O'Reilly  |
| &gt; 10:30    | ture]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
+------------+--------+-------------------------+---------------------+
| 10:30 -    | &gt;      |                         | &gt; -                |
| 11:00      | Coffee |                         |                     |
|            | &gt;      |                         |                     |
|            |  Break |                         |                     |
|            | &gt; and  |                         |                     |
|            | &gt; Q&amp;A  |                         |                     |
+------------+--------+-------------------------+---------------------+
| 11:00 -    | &gt; [Lec |                         | &gt; Dr Paul O'Reilly  |
| 12:00      | ture]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | &gt;      |                         |                     |
|            | Introd |                         |                     |
|            | uction |                         |                     |
|            | &gt; to   |                         |                     |
|            | &gt; PRS  |                         |                     |
|            | &gt; II   |                         |                     |
+------------+--------+-------------------------+---------------------+
| 12:00 -    | &gt;      |                         | &gt; -                |
| 13:30      |  Lunch |                         |                     |
+------------+--------+-------------------------+---------------------+
| 13:30 -    | &gt;      |                         | &gt; Dr Conrad Iyegbe  |
| 14:30      | [Pract |                         | &gt; &amp; Tutors          |
|            | ical]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | &gt;      |                         |                     |
|            | Introd |                         |                     |
|            | uction |                         |                     |
|            | &gt; to   |                         |                     |
|            | &gt; PRS  |                         |                     |
|            | &gt; I    |                         |                     |
+------------+--------+-------------------------+---------------------+
| 14:30 -    | &gt;      |                         | &gt; -                |
| 15:00      | Coffee |                         |                     |
|            | &gt;      |                         |                     |
|            |  Break |                         |                     |
|            | &gt; and  |                         |                     |
|            | &gt; Q&amp;A  |                         |                     |
+------------+--------+-------------------------+---------------------+
| 15:00 -    | &gt;      |                         | &gt; Dr Conrad Iyegbe  |
| 16:00      | [Pract |                         | &gt; &amp; Tutors          |
|            | ical]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | &gt;      |                         |                     |
|            | Introd |                         |                     |
|            | uction |                         |                     |
|            | &gt; to   |                         |                     |
|            | &gt; PRS  |                         |                     |
|            | &gt; II   |                         |                     |
+------------+--------+-------------------------+---------------------+
| 16:00 -    | &gt; [S   |                         | &gt; Dr Nicki Tiffin   |
| 17:00      | pecial |                         |                     |
|            | &gt; Sem  |                         |                     |
|            | inar]{ |                         |                     |
|            | .under |                         |                     |
|            | line}: |                         |                     |
|            | &gt; Pol  |                         |                     |
|            | ygenic |                         |                     |
|            | &gt; Risk |                         |                     |
|            | &gt;      |                         |                     |
|            | Scores |                         |                     |
|            | &gt;      |                         |                     |
|            | &gt; +   |                         |                     |
|            | &gt; and  |                         |                     |
|            | &gt;      |                         |                     |
|            | Ethics |                         |                     |
+------------+--------+-------------------------+---------------------+</p>
<blockquote>
<p><strong>Contents</strong></p>
</blockquote>
<p><a href="#day-2-timetable">Day 2 Timetable</a> 1</p>
<p><a href="#day-2-timetable">Day 2 Timetable</a> . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . 1</p>
<ol>
<li>
<p><a href="#introduction-to-polygenic-score-analyses">Introduction to Polygenic Score
    &gt; Analyses</a> 3</p>
<ol>
<li>
<p><a href="#key-learning-outcomes">Key Learning Outcomes</a> . . . . . . . .
    . . . . . . . . . . . . . . . . 3</p>
</li>
<li>
<p><a href="#resources-you-will-be-using">Resources you will be using</a> . .
    . . . . . . . . . . . . . . . . . . . . 3</p>
</li>
<li>
<p><a href="#data-structure">Data Structure</a> . . . . . . . . . . . . . . .
    . . . . . . . . . . . . . . 4</p>
</li>
<li>
<p><a href="#introduction">Introduction</a> . . . . . . . . . . . . . . . . .
    . . . . . . . . . . . . . . 5</p>
</li>
<li>
<p><a href="#understanding-gwas-summary-statistics">Understanding GWAS Summary
    Statistics</a> . . . . . .
    . . . . . . . . 5</p>
</li>
<li>
<p><a href="#matching-the-base-and-target-data-sets">Matching the Base and Target Data
    sets</a> . . . . . . . . .
    . . . . . . 6</p>
</li>
<li>
<p><a href="#linkage-disequilibrium-in-prs-analyses">Linkage Disequilibrium in PRS
    Analyses</a> . . . . . . .
    . . . . . . . . 7</p>
<ol>
<li><a href="#performing-clumping">Performing Clumping</a> . . . . . . . .
    . . . . . . . . . . . . . . 7</li>
</ol>
</li>
<li>
<p><a href="#p-value-thresholding">P-Value Thresholding</a> . . . . . . . . .
    . . . . . . . . . . . . . . . . . 8</p>
<ol>
<li>
<p><a href="#height-prs-using-gw-significant-snps-only">Height PRS using GW-significant SNPs
    only</a> . . . . .
    . . . . 8</p>
</li>
<li>
<p><a href="#height-prs-across-multiple-p-value-thresholds">Height PRS across multiple P-value
    thresholds</a>
    10</p>
</li>
<li>
<p><a href="#high-resolution-scoring">High Resolution Scoring</a> 12</p>
</li>
</ol>
</li>
<li>
<p><a href="#stratifying-samples-by-prs">Stratifying Samples by PRS</a> 15</p>
</li>
<li>
<p><a href="#case-control-studies">Case Control Studies</a> 17</p>
</li>
<li>
<p><a href="#cross-trait-analysis">Cross-Trait Analysis</a> 18</p>
</li>
</ol>
</li>
</ol>
<h1 id="introduction-to-polygenic-score-analyses">Introduction to Polygenic Score Analyses</h1>
<h2 id="key-learning-outcomes">Key Learning Outcomes</h2>
<blockquote>
<p>After completing this practical, you should be able to:</p>
</blockquote>
<ol>
<li>
<p>Perform basic Polygenic Risk Score (PRS) analyses using PRSice
    (Euesden, Lewis &amp; O'Reilly 2015; Choi &amp; O'Reilly 2019)</p>
</li>
<li>
<p>Interpret results generated from PRS analyses</p>
</li>
<li>
<p>Customise visualisation of results</p>
</li>
</ol>
<h2 id="resources-you-will-be-using">Resources you will be using</h2>
<blockquote>
<p>To perform PRS analyses, summary statistics from Genome Wide
Association Stud- ies (GWAS) are required. In this workshop, the
following summary statistics are used:</p>
<p>Phenotype Provider Description Download Link</p>
<p>Height <a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files">GIANT
Consortium</a>
GWAS of height on</p>
<p>253,288 individuals (<strong>wood_defining_2014</strong>)</p>
<p><a href="https://portals.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz">Download</a>
<a href="https://portals.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz">Link</a></p>
<p>Coronary artery disease (CAD)</p>
<p><a href="http://www.cardiogramplusc4d.org/">CARDIoGRAM plus</a> <a href="http://www.cardiogramplusc4d.org/">C4D
consortium</a></p>
<p>GWAS on 60,801</p>
<p>CAD cases and 123,504 controls</p>
<p><a href="http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip">Download</a>
<a href="http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip">Link</a></p>
</blockquote>
<h4 class="unnumbered" id="consortium_comprehensive_2015">(consortium_comprehensive_2015)</h4>
<h2 id="data-structure">Data Structure</h2>
<blockquote>
<p>You will find all practical materials in the <strong>PRS_Workshop/Day_2</strong>
directory. Relevant materials that you should see there at the start
of the practical are as follows:</p>
<p><img alt="" src="media/image2.png" />Practical Base_Data</p>
<p>GIANT_Height.txt cad.add.txt cad.add.readme</p>
<p>Target_Data</p>
<p>TAR.fam TAR.bim TAR.bed TAR.height</p>
<p>TAR.cad TAR.covariate</p>
<p>Software</p>
<p>plink_mac plink_linux plink.exe</p>
<p>PRSice.R PRSice_mac PRSice_linux PRSice_win64.exe</p>
</blockquote>
<p><img alt="" src="media/image5.jpeg" />{width="0.3229166666666667in"
height="0.3229166666666667in"}</p>
<h2 id="introduction">Introduction</h2>
<blockquote>
<p>A PRS is a (usually weak) estimate of an individual's genetic
propensity to a pheno- type, calculated as a sum of their genome-wide
genotypes weighted by corresponding genotype effect sizes obtained
from GWAS summary statistics. In the next section we will consider
what the effect size means and how it is used in computing PRS.</p>
</blockquote>
<h2 id="understanding-gwas-summary-statistics">Understanding GWAS Summary Statistics</h2>
<blockquote>
<p>When GWAS are performed on a quantitative trait, the effect size is
typically given as a beta coefficient (ùõΩ) from a linear regression
with Single Nucleotide Polymor- phism (SNP) genotypes as predictor of
phenotype. The ùõΩ coefficient estimates the increase in the phenotype
for each copy of the <em>effect allele</em>. For example, if the effect
allele of a SNP is <strong>G</strong> and the non-effect allele is <strong>A</strong> , then the
genotypes <strong>AA</strong>, <strong>AG</strong> and <strong>GG</strong> will be coded as 0, 1 and 2
respectively. In this scenario, the ùõΩ coefficient reflects how much
the phenotype changes for each <strong>G</strong> allele present (NB. The ùõΩ can be
positive or negative - so the 'effect allele' is simply the allele
that was coded in the regression, not necessarily the allele with a
positive effect).</p>
<p>When a GWAS is performed on a binary trait (e.g. case-control study),
the effect size is usually reported as an Odd Ratios (OR). Using the
same example, if the OR from the GWAS is 2 with respect to the <strong>G</strong>
allele, then the OR of <strong>AG</strong> relative to <strong>AA</strong> is 2, and the OR of
<strong>GG</strong> relative to <strong>AA</strong> is 4. So an individual with the <strong>GG</strong>
genotype are estimated* to be 4 times more likely to be a case than
someone with the <strong>AA</strong> genotype (*an Odds Ratio is itself an
estimate of a Risk Ratio, which cannot be calculated from a
case/control study)</p>
</blockquote>
<p><img alt="" src="media/image6.jpeg" />{width="0.3281244531933508in"
height="0.3281244531933508in"}</p>
<blockquote>
<p><img alt="" src="media/image7.jpeg" />{width="0.3541655730533683in"
height="0.3593744531933508in"}</p>
</blockquote>
<h2 id="matching-the-base-and-target-data-sets">Matching the Base and Target Data sets</h2>
<blockquote>
<p>The first step in PRS calculation is to ensure consistency between the
GWAS sum- mary statistic file (<em>base data</em>) and the target genotype
file (<em>target data</em>). Since the base and target data are generated
independently, they often relate to different SNPs</p>
<p>- and so the first job is to identify the overlapping SNPs across the
two data sets and remove non-overlapping SNPs (this is usually done
for you by PRS software). If the overlap is low then it would be a
good idea to perform imputation on your target data to increase the
number of SNPs that overlap between the data sets.</p>
<p>The next, more tricky issue, is that the genotype encoding between the
data sets may differ. For example, while the effect allele of a SNP is
<strong>T</strong> in the base data, the effect allele in the target might be <strong>G</strong>
instead. When this occurs, <em>allele flipping</em> should be performed,
where the genotype encoding in the target data is reversed so that
<strong>TT</strong>, <strong>TG</strong> and <strong>GG</strong> are coded as 2, 1 and 0. Again, this is
usually performed automatically by PRS software.</p>
</blockquote>
<p><img alt="" src="media/image5.jpeg" />{width="0.3229166666666667in"
height="0.3229166666666667in"}</p>
<h2 id="linkage-disequilibrium-in-prs-analyses">Linkage Disequilibrium in PRS Analyses</h2>
<blockquote>
<p>GWAS are typically performed one-SNP-at-a-time, which, combined with
the strong correlation structure across the genome (Linkage
Disequilibrium (LD)), makes iden- tifying the independent genetic
effects (or their best proxies if these are not geno- typed/imputed)
challenging. There are two main options for approximating the PRS that
would have been generated from full conditional GWAS: 1. SNPs are
<em>clumped</em> so that the retained SNPs are largely independent of each
other, allowing their effects to be summed, assuming additive effects,
2. all SNPs are included and the LD between them is accounted for.</p>
<p>While option 2 is statistically appealing, option 1 has been most
adopted in PRS studies so far, most likely due to its simplicity and
the similarity of results of meth- ods using the different options to
date (<strong>mak_polygenic_2017</strong>). In this work- shop we will consider
option 1, implemented in PRSice, but if you are interested in how LD
can be incorporated as a parameter in PRS calculation then see the LD-
pred (<strong>vilhjalmsson_modeling_2015</strong>) and lassosum
(<strong>mak_polygenic_2017</strong>) papers.</p>
</blockquote>
<h3 id="performing-clumping">Performing Clumping</h3>
<blockquote>
<p><em>Clumping</em> is the procedure where a SNP data set is 'thinned' by
removing SNPs across the genome that are correlated (in high LD) with
a nearby SNP that has a smaller association ùëÉ -value.</p>
<p>SNPs are first sorted (i.e. ranked) by their ùëÉ -values. Then, starting
from the most significant SNP (denoted as the <em>index SNP</em>), any SNPs
in high LD (eg. ùëü<sup>2</sup> &gt; 0.1, with ùëü<sup>2</sup> typically calculated from
ùëù‚Ñéùëéùë†ùëíùëë ‚Ñéùëéùëùùëôùëúùë°ùë¶ùëùùëí data) with the index SNP are removed. To reduce
computational burden, only SNPs that are within e.g. 250 kb of the
<em>index SNP</em> are ùëêùëôùë¢ùëöùëùùëíùëë. This process is continued until no <em>index
SNPs</em> remain.</p>
<p>Use the command below to perform clumping of the Height GWAS data
using PLINK(<strong>chang_second_2015</strong>). First, you will have to navigate
to the right folder where the data are stored using the terminal. Open
the terminal and type the command below at the terminal prompt:</p>
<p>1 cd \~/Desktop/PRS\_Workshop/</p>
<p>Next type the following command (NB. See warning below):</p>
<p>1 ./Software/plink_linux</p>
<p>2 --bfile Target_Data/TAR</p>
<p>3 --clump Base_Data/GIANT_Height.txt</p>
<p>4 --clump-p1 1</p>
<p>5 --clump-snp-field MarkerName</p>
<p>6 --clump-field p</p>
<p>7 --clump-kb 250</p>
<p>8 --clump-r2 0.1</p>
<p>9 --out Results/Height</p>
</blockquote>
<p><img alt="" src="media/image5.jpeg" />{width="0.3229155730533683in"
height="0.3229166666666667in"}</p>
<blockquote>
<p>The command above performs clumping on the height GWAS using LD
calculated based on the <strong>TAR</strong> genotype file. SNPs that have ùëü<sup>2</sup> &gt;
0.1 within a 250 kb window of the index SNP are removed. This will
generate the <strong>Height.clumped</strong> file, which contains the SNPs retained
after clumping.</p>
</blockquote>
<p><img alt="" src="media/image7.jpeg" />{width="0.3541655730533683in"
height="0.3593744531933508in"}</p>
<h2 id="p-value-thresholding">P-Value Thresholding</h2>
<blockquote>
<p>Deciding which SNPs to include in the calculation of PRS is one of the
major challenges in the field. A simple and popular approach is to
include SNPs according to their GWAS association ùëÉ -value. For
example, we may choose to include only the genome-wide significant
SNPs from the GWAS because those are the SNPs with significant
evidence for association. In the next subsection you will compute PRS
from GW-significant SNPs only, and then in the subsequent subsection
you will generate multiple PRSs using different ùëÉ -value thresholds.</p>
</blockquote>
<h3 id="height-prs-using-gw-significant-snps-only">Height PRS using GW-significant SNPs only</h3>
<blockquote>
<p>Use the commands below to run PRSice with GIANT Height GWAS as base
data and the height phenotype target data. PRSice will calculate
Height PRS in the</p>
<p>target data and then perform a regression of the Height PRS against
the target individual's true height values. From the
<strong>PRS_Workshop/Day_2</strong> directory, run the following command in the
terminal:</p>
<p>1 Rscript ./Software/PRSice.R</p>
<p>2 --prsice Software/PRSice_linux</p>
<p>3 --base Base_Data/GIANT_Height.txt</p>
<p>4 --target Target_Data/TAR</p>
<p>5 --snp MarkerName</p>
<p>6 --A1 Allele1</p>
<p>7 --A2 Allele2</p>
<p>8 --stat b</p>
<p>9 --beta</p>
<p>10 --pvalue p</p>
<p>11 --pheno Target_Data/TAR.height</p>
<p>12 --binary-target F</p>
<p>13 --bar-levels 5e-8</p>
<p>14 --no-full</p>
<p>15 --fastscore</p>
<p>16 --out Results/Height.gws</p>
<p>This command takes the Height GWAS summary statistic file (--base),
informs PRSice of the column name for the column containing the SNP ID
(--snp), the effect allele (--A1), the non-effect allele (--A2),
the effect size (--stat) and the ùëÉ -value (--pvalue). We also inform
PRSice that the effect size is a ùõΩ coefficient (--beta) instead of an
OR. The --binary-target F command informs PRSice that the target
phenotype is a quantitative trait and thus linear regression should be
performed. In addi- tion, we ask PRSice not to perform high-resolution
scoring over multiple thresholds</p>
<p>(--fastscore), and to compute the PRS using only those SNPs with ùëÉ
-value \&lt; 5*√ó*10<sup>*‚àí*8</sup>.</p>
</blockquote>
<p><img alt="" src="media/image6.jpeg" />{width="0.3281244531933508in"
height="0.3281244531933508in"}</p>
<blockquote>
<p>PRSice performs strand flipping and clumping automatically and
generates the <strong>Height.gws.summary</strong> file, together with other output
that we will look into later in the practical. The summary file
contains the following columns:</p>
</blockquote>
<ol>
<li>
<p><strong>Phenotype</strong> - Name of Phenotype.</p>
</li>
<li>
<p><strong>Set</strong> - Name of Gene Set. Default is <em>Base</em></p>
</li>
<li>
<p><strong>Threshold</strong> - Best P-value Threshold</p>
</li>
<li>
<p><strong>PRS.R2</strong> - Variance explained by the PRS</p>
</li>
<li>
<p><strong>Full.R2</strong> - Variance explained by the full model (including the
    covariates)</p>
</li>
<li>
<p><strong>Null.R2</strong> - Variance explained by the covariates (none provided
    here)</p>
</li>
<li>
<p><strong>Prevalence</strong> - The population disease prevalence as indicated by
    the user (not provided here due to testing continuous trait)</p>
</li>
<li>
<p><strong>Coefficient</strong> - The ùõΩ coefficient corresponding to the effect
    estimate of the best-fit PRS on the target trait in the regression.
    A one unit increase in the PRS increases the outcome by ùõΩ</p>
</li>
<li>
<p><strong>Standard.Error</strong> - The standard error of the best-fit PRS ùõΩ
    coefficient (see above)</p>
</li>
<li>
<p><strong>P</strong> - The ùëÉ -value relating to testing the null hypothesis that
    the best-fit PRS</p>
</li>
</ol>
<blockquote>
<p>ùõΩ coefficient is zero.</p>
</blockquote>
<ol>
<li>
<p><strong>Num_SNP</strong> - Number of SNPs included in the best-fit PRS</p>
</li>
<li>
<p><strong>Empirical-P</strong> - Only provided if permutation is performed. This is
    the em- pirical ùëÉ -value corresponding to the association test of
    the best-fit PRS - this controls for the over-fitting that occurs
    when multiple thresholds are tested.</p>
</li>
</ol>
<blockquote>
<p>For now, we can ignore most columns and focus on the <strong>PRS.R2</strong> and
the <strong>P</strong> column, which provide information on the model fit.</p>
</blockquote>
<p><img alt="" src="media/image7.jpeg" />{width="0.3541666666666667in"
height="0.3593744531933508in"}</p>
<h3 id="height-prs-across-multiple-p-value-thresholds">Height PRS across multiple P-value thresholds</h3>
<blockquote>
<p>A disadvantage of using only genome-wide significant SNPs is that
there are likely to be many true signals among SNPs that did not reach
genome-wide significance. However, since we do not know what ùëÉ -value
threshold provides the \"best\" predic- tion for our particular data,
then we can calculate the PRS under several ùëÉ -value thresholds and
test their prediction accuracy to identify the \"best\" threshold (NB.</p>
<p>[]{#_bookmark13 .anchor}Figure 1.1: BARPLOT generated by PRSice</p>
</blockquote>
<p><img alt="" src="media/image8.png" />{width="2.7440616797900264in"
height="2.761874453193351in"}</p>
<blockquote>
<p>See <strong>dudbridge_power_2013</strong> for theory on factors affecting the
best-fit PRS). This process is implemented in PRSice and can be
performed automatically as fol- lows:</p>
<p>1 Rscript ./Software/PRSice.R</p>
<p>2 --dir .</p>
<p>3 --prsice Software/PRSice_linux</p>
<p>4 --base Base_Data/GIANT_Height.txt</p>
<p>5 --target Target_Data/TAR</p>
<p>6 --snp MarkerName</p>
<p>7 --A1 Allele1</p>
<p>8 --A2 Allele2</p>
<p>9 --stat b</p>
<p>10 --beta</p>
<p>11 --pvalue p</p>
<p>12 --pheno Target_Data/TAR.height</p>
<p>13 --binary-target F</p>
<p>14 --fastscore</p>
<p>15 --out Results/Height.fast</p>
<p>By removing the --bar-levels and --no-full command, we ask PRSice to
perform PRS calculation with a number of predefined thresholds (0.001,
0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 1). The <strong>.prsice</strong> file is very
similar to the <strong>.summary</strong> file, the only difference is that</p>
<p><strong>.prsice</strong> file reports the results of model fits for <strong>all
thresholds</strong> instead of the most predictive threshold. This allow us
to observe the change in model fitting across different thresholds,
visualized by the BARPLOT (fig. <a href="#_bookmark13">1.1)</a> generated by
PRSice</p>
<p><img alt="" src="media/image7.jpeg" />{width="0.3541655730533683in"
height="0.3593744531933508in"}</p>
</blockquote>
<h3 id="high-resolution-scoring">High Resolution Scoring</h3>
<blockquote>
<p>If we limit ourselves to a small number of ùëÉ -value thresholds, we
might \"miss\" the most predictive threshold. In order to identify
this \"best\" threshold, we will need \"high-resolution scoring\",
that is, to test the predictive power of PRS generated under a large
number of p-value thresholds. We can achieve that by simply removing
the --fastscore command from the PRSice script:</p>
<p>1 Rscript ./Software/PRSice.R</p>
<p>2 --dir .</p>
<p>3 --prsice Software/PRSice_linux</p>
<p>4 --base Base_Data/GIANT_Height.txt</p>
<p>5 --target Target_Data/TAR</p>
<p>6 --snp MarkerName</p>
<p>7 --A1 Allele1</p>
<p>8 --A2 Allele2</p>
<p>9 --stat b</p>
<p>10 --beta</p>
<p>11 --pvalue p</p>
<p>12 --pheno Target_Data/TAR.height</p>
<p>13 --binary-target F</p>
<p>14 --out Results/Height.highres</p>
<p>When PRSice performs high-resolution scoring, it will generate a plot
(fig. <a href="#_bookmark15">1.2)</a> presenting the model fit of PRS calculated
at all P-value thresholds.</p>
<p>[]{#_bookmark15 .anchor}Figure 1.2: High Resolution Plot generated by
PRSice</p>
</blockquote>
<p><img alt="" src="media/image9.jpeg" />{width="2.463332239720035in"
height="2.455in"}<img alt="" src="media/image7.jpeg" />{width="0.3541655730533683in"
height="0.3593744531933508in"}</p>
<blockquote>
<p><strong>Accounting for Covariates</strong></p>
<p>When performing PRS, one might want to account for covariates. Based
on user in- puts, PRSice can automatically incorporate covariates into
its model. For example, the following commands will include sex into
the regression model as a covariate:</p>
<p>1 Rscript ./Software/PRSice.R</p>
<p>2 --prsice Software/PRSice_linux</p>
<p>3 --base Base_Data/GIANT_Height.txt</p>
<p>4 --target Target_Data/TAR</p>
<p>5 --snp MarkerName</p>
<p>6 --A1 Allele1</p>
<p>7 --A2 Allele2</p>
<p>8 --stat b</p>
<p>9 --beta</p>
<p>10 --pvalue p</p>
<p>11 --pheno Target_Data/TAR.height</p>
<p>12 --binary-target F</p>
<p>13 --cov Target_Data/TAR.covariate</p>
<p>14 --cov-col Sex</p>
<p>15 --out Results/Height.sex</p>
<p>When covariates were include in the analysis, PRSice will use the
model fit of <strong>only PRS</strong> for all its output. This ùëÉ ùëÖùëÜ.ùëÖ<sup>2</sup> is
calculated by minusing the</p>
<p>ùëÖ<sup>2</sup> of the null model (e.g. ùêªùëíùëñùëî‚Ñéùë° <em>‚àº</em> ùëÜùëíùë•) from the ùëÖ<sup>2</sup> of the full
model (e.g.</p>
<p><img alt="" src="media/image10.jpeg" />{width="0.34781167979002625in"
height="0.34781167979002625in"}<img alt="" src="media/image7.jpeg" />{width="0.3541666666666667in"
height="0.3593744531933508in"}ùêªùëíùëñùëî‚Ñéùë° <em>‚àº</em> ùëÜùëíùë• + ùëÉ ùëÖùëÜ).</p>
<p>Usually, with categorical variables, dummy variables have to be
generated to rep- resent the different categories. Alternatively,
residualized phenotype, generated by regresing the covariates against
the phenotype, can be used for downstream anal- yses. A useful feature
of PRSice is to automatically generate the dummy variable for users.
This can be achieved with the following command:</p>
<p>1 Rscript.exe ./Software/PRSice.R</p>
<p>2 --prsice Software/PRSice_linux</p>
<p>3 --base Base_Data/GIANT_Height.txt</p>
<p>4 --target Target_Data/TAR</p>
<p>5 --snp MarkerName</p>
<p>6 --A1 Allele1</p>
<p>7 --A2 Allele2</p>
<p>8 --stat b</p>
<p>9 --beta</p>
<p>10 --pvalue p</p>
<p>11 --pheno Target_Data/TAR.height</p>
<p>12 --binary-target F</p>
<p>13 --cov Target_Data/TAR.covariate</p>
<p>14 --cov-col Sex</p>
<p>15 --cov-factor Sex</p>
<p>16 --out Results/Height.sex</p>
<p>[]{#_bookmark16 .anchor}Figure 1.3: Example of a quantile plot
generated by PRSice</p>
</blockquote>
<p><img alt="" src="media/image11.png" />{width="2.4641655730533683in"
height="2.4674989063867017in"}</p>
<h2 id="stratifying-samples-by-prs">Stratifying Samples by PRS</h2>
<blockquote>
<p>An interesting application of PRS is to test whether samples with
higher PRS have higher phenotypic values. This can be nicely
visualized using the quantile plot (fig. <a href="#_bookmark16">1.3).</a></p>
<p>To generate quantile plots in PRSice, simply add --quantile 10
option.</p>
</blockquote>
<p><img alt="" src="media/image6.jpeg" />{width="0.3281244531933508in"
height="0.3281244531933508in"}</p>
<blockquote>
<p>1 Rscript ./Software/PRSice.R</p>
<p>2 --prsice Software/PRSice_linux</p>
<p>3 --base Base_Data/GIANT_Height.txt</p>
<p>4 --target Target_Data/TAR</p>
<p>5 --snp MarkerName</p>
<p>6 --A1 Allele1</p>
<p>7 --A2 Allele2</p>
<p>8 --stat b</p>
<p>9 --beta</p>
<p>10 --pvalue p</p>
<p>11 --pheno Target_Data/TAR.height</p>
<p>12 --binary-target F</p>
<p>13 --cov Target_Data/TAR.covariate</p>
<p>14 --cov-col Sex</p>
<p>15 --plot</p>
<p>16 --quantile 10</p>
<p>17 --out Results/Height.sex</p>
</blockquote>
<p><img alt="" src="media/image10.jpeg" />{width="0.34781167979002625in"
height="0.34781167979002625in"}</p>
<blockquote>
<p>Figure 1.4: Example of a strata plot generated by PRSice</p>
</blockquote>
<p><img alt="" src="media/image12.jpeg" />{width="2.4641655730533683in"
height="2.4674989063867017in"}</p>
<blockquote>
<p>A disadvantage of the quantile plot is that it only seperate samples
into quantiles of equal size. However, it is somtimes interesting to
investigate whether a specific strata (e.g. top 5% of samples),
contain a higher PRS than the reference strata. For example,
<strong>mavaddat_prediction_2015</strong> found that samples in the highest 1% of
PRS distribution have a 2.81 increased OR of breast cancer when
comparing to samples at the middle quantiles (40th to 60th
percentile). We can mimic their table by using --quant-break, which
represents the upper bound of each strata, and --quant-ref, which
represents the upper bound of the reference quantile:</p>
<p>1 Rscript ./Software/PRSice.R</p>
<p>2 --prsice Software/PRSice_linux</p>
<p>3 --base Base_Data/GIANT_Height.txt</p>
<p>4 --target Target_Data/TAR</p>
<p>5 --snp MarkerName</p>
<p>6 --A1 Allele1</p>
<p>7 --A2 Allele2</p>
<p>8 --stat b</p>
<p>9 --beta</p>
<p>10 --pvalue p</p>
<p>11 --pheno Target_Data/TAR.height</p>
<p>12 --binary-target F</p>
<p>13 --cov Target_Data/TAR.covariate</p>
<p>14 --cov-col Sex</p>
<p>15 --plot</p>
<p>16 --quantile 100</p>
<p>17 --quant-break 1,5,10,20,40,60,80,90,95,99,100</p>
<p>18 --quant-ref 60</p>
<p>19 --out Results/Height.sex</p>
</blockquote>
<p><img alt="" src="media/image6.jpeg" />{width="0.3281244531933508in"
height="0.3281244531933508in"}</p>
<h2 id="case-control-studies">Case Control Studies</h2>
<blockquote>
<p>In the previous exercises, we have performed PRS analyses on height,
which is a quantitative trait. For binary phenotypes (e.g
case-control) there are a number of differences in the analysis:</p>
</blockquote>
<ol>
<li>
<p>Logistic regression has to be performed instead of linear regression</p>
</li>
<li>
<p>ORs are usually provided and need to be converted to ùõΩ's when
    constructing PRS</p>
</li>
</ol>
<blockquote>
<p>Here we will use CAD as an example. You will find the summary
statistic under <em>Base_Data</em> (<strong>cad.add.txt</strong>) and the phenotype file
(<strong>TAR.cad</strong>) under <em>Tar- get_Data</em>. You will also need to specify
--binary-target T in the PRSice command to indicate that the
phenotype is binary.</p>
</blockquote>
<p><img alt="" src="media/image10.jpeg" />{width="0.34781167979002625in"
height="0.34781167979002625in"}</p>
<blockquote>
<p>1 Rscript ./Software/PRSice.R</p>
<p>2 --prsice Software/PRSice_linux</p>
<p>3 --base Base_Data/cad.add.txt</p>
<p>4 --target Target_Data/TAR</p>
<p>5 --snp markername</p>
<p>6 --A1 effect_allele</p>
<p>7 --A2 noneffect_allele</p>
<p>8 --chr chr</p>
<p>9 --bp bp_hg19</p>
<p>10 --stat beta</p>
<p>11 --beta</p>
<p>12 --pvalue p_dgc</p>
<p>13 --pheno Target_Data/CAD.pheno</p>
<p>14 --binary-target T</p>
<p>15 --out Results/CAD.highres</p>
</blockquote>
<p><img alt="" src="media/image10.jpeg" />{width="0.34781167979002625in"
height="0.34781167979002625in"}</p>
<blockquote>
<p><img alt="" src="media/image13.jpeg" />{width="3.5316666666666667in"
height="2.066457786526684in"}</p>
<p>[]{#_bookmark19 .anchor}Figure 1.5: Plot taken from Ruderfer et al.
2014</p>
</blockquote>
<p><img alt="" src="media/image7.jpeg" />{width="0.3541655730533683in"
height="0.3593744531933508in"}</p>
<h2 id="cross-trait-analysis">Cross-Trait Analysis</h2>
<blockquote>
<p>A popular application of PRS is in performing cross-trait analyses.
This allows some interesting analyses such as those performed by
<strong>ruderfer_polygenic_2014</strong> (fig. <a href="#_bookmark19">1.5),</a> which used the
bipolar PRS to predict into different clinical dimensions of
schizophrenia.</p>
<p>In this practical, we will perform cross-trait analyses between CAD
and Height, using height as the base and CAD as the target.</p>
</blockquote>
<p><img alt="" src="media/image10.jpeg" />{width="0.34781167979002625in"
height="0.34781167979002625in"}</p>
<blockquote>
<p>1 Rscript ./Software/PRSice.R</p>
<p>2 --prsice Software/PRSice_linux</p>
<p>3 --base Base_Data/GIANT_Height.txt</p>
<p>4 --target Target_Data/TAR</p>
<p>5 --snp MarkerName</p>
<p>6 --A1 Allele1</p>
<p>7 --A2 Allele2</p>
<p>8 --stat b</p>
<p>9 --beta</p>
<p>10 --pvalue p</p>
<p>11 --pheno Target_Data/CAD.pheno</p>
<p>12 --binary-target T</p>
<p>13 --out Results/Cross.highres</p>
</blockquote>
<p><img alt="" src="media/image7.jpeg" />{width="0.3541655730533683in"
height="0.3593744531933508in"}</p>
              
            </div>
          </div><footer>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/tadesouaiaia/sibArc-website" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
    
  </span>
</div>
    <script src="../../../../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../../../../..";</script>
    <script src="../../../../../js/theme_extra.js"></script>
    <script src="../../../../../js/theme.js"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      <script src="../../../../../javascripts/details.js"></script>
      <script src="../../../../../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
