<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="author" content="Tade Souaiaia" />
      <link rel="shortcut icon" href="../../../../img/favicon.ico" />
    <title>Advanced Polygenic Risk Score Analyses - SibArc</title>
    <link rel="stylesheet" href="../../../../css/theme.css" />
    <link rel="stylesheet" href="../../../../css/theme_extra.css" />
        <link href="../../../../css/extra.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Advanced Polygenic Risk Score Analyses";
        var mkdocs_page_input_path = "tmp/prs_2023-main/modules/Day3.docx.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../../../js/html5shiv.min.js"></script>
    <![endif]--> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../../../.." class="icon icon-home"> SibArc
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../../..">Home</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Quick Start</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../quik_install/">Installation and Data</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../quik_demo/">Running SibArc</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../quik_result/">Interpreting Results</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">About</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../about_future/">Future Work</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Misc</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../../workshop_practical_paul.docx/">Day 1 (AM)</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../workshop_practical_tade/">Day 1 (PM)</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../../../workshop_practical_clive.docx/">Day 2 (AM)</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../..">SibArc</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../../../.." class="icon icon-home" aria-label="Docs"></a></li>
      <li class="breadcrumb-item active">Advanced Polygenic Risk Score Analyses</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/tadesouaiaia/sibArc-website/edit/master/docs/tmp/prs_2023-main/modules/Day3.docx.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="advanced-polygenic-risk-score-analyses">Advanced Polygenic Risk Score Analyses</h1>
<h2 id="day-3-polygenic-risk-score-analyses-workshop-2023">Day 3 - Polygenic Risk Score Analyses Workshop 2023</h2>
<h2 id="table-of-contents">Table of Contents</h2>
<ol>
<li><a href="#key-learning-outcomes">Key Learning Outcomes</a></li>
<li><a href="#resources-you-will-be-using">Resources you will be using</a></li>
<li><a href="#data-sets">Datasets</a> </li>
<li><a href="#exercise-1-estimating-r2-in-case-and-control-studies">Exercise 1 Estimating R<sup>2</sup> in case and control studies</a></li>
<li><a href="#exercise-2-Overfitting-caused-by-model-optimisation">Exercise 2 Overfitting caused by model optimisation</a><ol>
<li><a href="#out-of-sample-validation">Out of Sample Validation</a></li>
</ol>
</li>
<li><a href="#exercise-3-distribution-of-prs">Exercise 3 Distribution of PRS</a></li>
<li><a href="#gene-set-analysis">Gene Set Analysis</a><ol>
<li><a href="#molecular-signatures-Database-msigdb">Molecular Signatures Database MSigDB</a></li>
<li><a href="#general-transfer-format-file">General Transfer Format file</a></li>
<li><a href="#browser-extensible-data-bed">Browser Extensible Data BED</a></li>
</ol>
</li>
<li><a href="#gene-set-enrichment-analysis">Gene Set Enrichment Analysis</a></li>
<li><a href="#exercise-4-gene-set-based-prs-analysis">Exercise 4 Gene Set Based PRS Analysis</a></li>
</ol>
<h2 id="key-learning-outcomes">Key Learning Outcomes</h2>
<p>After completing this practical, you should be able to:
  1. know how to adjust for ascertainment bias in case-control analysis
  2. Know how over-fitting aÔ¨Äects PRS results and how to handle it 
  3. understand distribution of PRS
  4. understand diÔ¨Äerent file formats involved in gene-set analysis
  5. understand diÔ¨Äerence between self-contained and competitive gene-set analyses
  6. Calculate pathway based PRS</p>
<h2 id="resources-you-will-be-using">Resources you will be using</h2>
<p>To perform PRS analyses, summary statistics from Genome-Wide Association Studies (GWAS) are required. In this workshop, the following summary statistics are used:</p>
<table>
<thead>
<tr>
<th style="text-align: center;"><strong>Phenotype</strong></th>
<th style="text-align: center;"><strong>Provider</strong></th>
<th style="text-align: center;"><strong>Description</strong></th>
<th style="text-align: center;"><strong>Download Link</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">Height</td>
<td style="text-align: center;"><a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files">GIANT Consortium</a></td>
<td style="text-align: center;">GWAS of height on 253,288 individuals</td>
<td style="text-align: center;"><a href="https://portals.broadinstitute.org/collaboration/giant/images/0/01/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt.gz">Link</a></td>
</tr>
<tr>
<td style="text-align: center;">Coronary artery disease (CAD)</td>
<td style="text-align: center;"><a href="http://www.cardiogramplusc4d.org/">CARDIoGRAMplusC4D Consortium</a></td>
<td style="text-align: center;">GWAS on 60,801 CAD cases and 123,504 controls</td>
<td style="text-align: center;"><a href="http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip">Link</a></td>
</tr>
</tbody>
</table>
<h2 id="additional-resources">Additional Resources</h2>
<p>In this practical, we will explore how to perform gene-set based PRS analyses. To perform this analysis, gene-set information and coordinates for the genic regions are required. These information can be obtained from the following database:</p>
<table>
<thead>
<tr>
<th style="text-align: center;"><strong>Data Set</strong></th>
<th style="text-align: center;"><strong>Description</strong></th>
<th style="text-align: center;"><strong>Download Link</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">Ensembl Human Genome GTF file</td>
<td style="text-align: center;">A file containing the coordinates for genes in the human genome. Used by PRSice to map the SNPs onto genic regions</td>
<td style="text-align: center;"><a href="https://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/">Link</a></td>
</tr>
<tr>
<td style="text-align: center;">MSigDB Gene Sets</td>
<td style="text-align: center;">File containing the gene-set information. <em>Free registration required.</em></td>
<td style="text-align: center;"><a href="http://software.broadinstitute.org/gsea/msigdb/download_file.jsp?filePath=/resources/msigdb/6.1/h.all.v6.1.symbols.gmt">Download here after registration</a></td>
</tr>
</tbody>
</table>
<h2 id="data-sets">Data Sets</h2>
<p>You will need to download the required files for this tutorial.       </p>
<p><a href="https://we.tl/t-YO8YbDjDK7">Download Backup WeTransfer</a>        </p>
<p>The data will be downloaded into your "Downloads" folder. You will need to move it to right directory, using the following commands.</p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>cd data
mv ~/Downloads/Day_3.zip .
unzip Day_3.zip
cd Day_3
</code></pre></div>
<p>You will find all practical materials in the <strong>data/Day_3</strong> directory. Relevant materials that you should see there at the start of the practical are as follows:</p>
<p>üìÇ: Base_Data
  - GIANT_Height.txt,
  - cad.add.txt,
  - cad.add.readme.</p>
<p>üìÇ: Target_Data
  - TAR.fam
  - TAR.bim
  - TAR.bed
  - TAR.height
  - TAR.cad 
  - TAR.covariate</p>
<p>üìÅ: Reference files
   - Homo_sapiens.GRCh38.86.gt
   - Sets.gmt</p>
<p>üõ†Ô∏è: Software
  - PRSice.R 
  - PRSice_linux
  - nagelkerke.R
  - Quantile.R</p>
<hr />
<blockquote>
<p>‚ÄºÔ∏è All target phenotype data in this worshop are <strong>simulated</strong>. They have no specific biological meaning and are for demonstration purposes only. 
</p>
</blockquote>
<hr />
<p><a href="#top"><a href="#table-of-contents">Back to Top</a></a></p>
<h2 id="exercise-1-estimating-r2-in-case-and-control-studies">Exercise 1 Estimating R<sup>2</sup> in case and control studies</h2>
<p>Bias in R<sup>2</sup> estimation caused by ascertained case/control samples can be adjusted using the equation proposed by <strong>Lee et al (2011)</strong>, which requires the sample prevalence (case/control ratio) and population prevalence as parameters. This function is implemented in PRSice and the adjustment can be performed by providing the population prevalence to the command <strong>--prevalence</strong>.</p>
<p>Residuals of logistic regression is not well defined, and in PRS analyses, Nagelkerke R<sup>2</sup> is usually used to represent the model R<sup>2</sup> (this is the default of PRSice). However, this R<sup>2</sup> does not account for the diÔ¨Äerence between sample prevalence (i.e. case-control ratio) and population prevalence, which can lead to bias in the reported R<sup>2</sup> (Figure 1.1a). </p>
<blockquote>
<p><strong>Figure 1.1: Performance of diÔ¨Äerent R<sup>2</sup> when the study contains equal portion of cases and controls</strong></p>
<p><strong>(a) Nagelkerke R<sup>2</sup></strong> 
<img alt="Figure 1.1a" src="https://drive.google.com/uc?id=1r3LV442RQT3CWGSAnZsU6QT1DqHtRKkd" /></p>
</blockquote>
<hr />
<p>Bias in R<sup>2</sup> estimation caused by ascertained case/control samples can be adjusted using the equation proposed by <strong>Lee et al. 2011 (Figure 1.1b)</strong>, which requires the sample
prevalence (case/control ratio) and population prevalence as parameters. This function is implemented in PRSice and the adjustment can be performed by providing the population prevalence to the command <strong>--prevalence</strong>.</p>
<blockquote>
<p><strong>Figure 1.1: Performance of diÔ¨Äerent R<sup>2</sup> when the study contains equal portion of cases and controls</strong></p>
<p><strong>(b) Lee adjusted R<sup>2</sup></strong> 
<img alt="Figure 1.1b" src="https://drive.google.com/uc?id=19ACkFb2gr7EU4dfcsTDPtprwtYPDQufo" /></p>
</blockquote>
<hr />
<p>Now, account for the ascertainment of the case/control sample by including the population prevalence (let‚Äôs assume e.g. 5% here) in the PRSice command to obtain the adjusted (Lee) R<sup>2</sup> :</p>
<p><div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
--prsice Software/PRSice_linux \
--base  Base_Data/cad.add.txt \
--target Target_Data/TAR \
--snp markername \
--A1 effect_allele \
--A2 noneffect_allele \
--chr chr \
--bp bp_hg19 \
--stat beta \
--beta \
--pvalue p_dgc \
--pheno Target_Data/TAR.cad \
--prevalence 0.05 \ 
--binary-target T \
--out Results/CAD.highres.LEER2
</code></pre></div>
The results are written to the "Results" directory. Examine the results folder and each file that was generated. For more information about each file type, see  <a href="https://choishingwan.github.io/PRSice/step_by_step/">here</a>.</p>
<hr />
<blockquote>
<p>‚≠ê Check the *.summary file in the Results folder where you will find the usual (Nagelkerke) R<sup>2</sup> and the adjusted (Lee) R<sup>2</sup>.
</p>
<p><strong>Figure 1.2: Barplot of CAD Lee R<sup>2</sup></strong> 
<img alt="Figure 1.2" src="https://drive.google.com/uc?id=1ktv_UydzcZXUdup9oTB1LKkFyhCT9WdY" /></p>
</blockquote>
<hr />
<hr />
<blockquote>
<p>üìå To speed up the practical, we have generated a smaller gene-set file. If you want the full gene-set file, you can download it from the link above.</p>
<p>üìå All target phenotype data in this workshop are simulated. While they reflect the corresponding trait data, they have no specific biological meaning and are for demonstration purposes only.</p>
</blockquote>
<hr />
<hr />
<blockquote>
<p>‚ùìHas accounting for the population prevalence aÔ¨Äected the R<sup>2</sup>?
<details>
<summary>Solution</summary>     </p>
<p>Yes, the adjusted R<sup>2</sup> = 0.0521524 and default R<sup>2</sup> = 0.0442664      </p>
</details>
</blockquote>
<hr />
<blockquote>
<p>‚ùìWould you expect a diÔ¨Äerence between the Nagelkerke R<sup>2</sup> and the Lee adjusted R<sup>2</sup> if the case/control ratio in the target sample reflects the disease prevalence in the population?</p>
<p><details>
<summary>Solution</summary>     </p>
<p>No, the R<sup>2</sup> will be the same because the prevalence of the target is a true representation of the population prevalence.      </p>
</details>
</blockquote>
<hr />
<p><a href="#top"><a href="#table-of-contents">Back to Top</a></a></p>
<h2 id="exercise-2-overfitting-caused-by-model-optimisation">Exercise 2 Overfitting caused by model optimisation</h2>
<p>In PRS analyses, the shrinkage or tuning parameter is usually optimized across a wide range of parametric space (e.g. P -value threshold, proportion of causal SNPs). When both optimisation and association testing are performed on the target data, over-fitted results will be obtained. The accuracy and predictive power of over-fitted results are likely to diminish when replicated in an independent data set.</p>
<p>A simple solution is to perform permutation to obtain an empirical P -value for the association model, which is implemented in PRSice. Briefly, permutation is performed as follows:
1) Compute the P -value in your original data, denoted as obs.p, at the "best" threshold.
2) Then shuÔ¨Ñe the phenotype and obtain the P -value of the "best" threshold for this null phenotype, denoted as null.p
3) Repeat 2) N times
4) Calculate the empirical P-value as:</p>
<h2 id="pemp-sumobsp-nullpi-1-n-1"><span class="arithmatex">\(<code>Pemp = (\sum(obs.p &gt; null.pi + 1) / (N + 1)</code>\)</span></h2>
<p>You will have to specify the number of permutation (N ) to perform by providing --perm N as a parameter to PRSice.</p>
<h2 id="rscript-softwareprsicer-prsice-softwareprsice_linux-base-base_datagiant_heighttxt-target-target_datatar-snp-markername-a1-allele1-a2-allele2-stat-b-beta-pvalue-p-pheno-target_datatarheight-binary-target-f-cov-target_datatarcovariate-cov-col-sex-perm-1000-out-resultsheightperm"><div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
    --prsice Software/PRSice_linux \
    --base  Base_Data/GIANT_Height.txt \
    --target Target_Data/TAR \
    --snp MarkerName \
    --A1 Allele1 \
    --A2 Allele2 \
    --stat b \
    --beta \
    --pvalue p \
    --pheno Target_Data/TAR.height \
    --binary-target F \
    --cov Target_Data/TAR.covariate \
    --cov-col Sex \
    --perm 1000 \
    --out Results/Height.perm
</code></pre></div></h2>
<blockquote>
<p><strong>Figure 1.3: Barplot of Height using 1000 permutations</strong></p>
<p><img alt="Figure 1.3" src="/images/day3/Height.perm_BARPLOT_2023-06-30.png" />
</p>
</blockquote>
<hr />
<hr />
<blockquote>
<p>üìù <strong>10000 permutations typically provide empirical P-values with high accuracy to the second decimal place (eg. 0.05), but smaller empirical P-values should be considered approximate.</strong>
</p>
</blockquote>
<hr />
<blockquote>
<p>‚ùì What is the smallest possible empirical P-value when 10000 permutation are performed? </p>
<p><details>
<summary>Solution</summary>     </p>
<p>1.5 X 10<sup>-34</sup>.      </p>
</details>
<p>‚ùì Is the height PRS significantly associated with height after accounting for the over-fitting implicit in identifying the best-fit PRS? How about CAD?</p>
<p><details>
<summary>Solution</summary>     </p>
<p>Yes, the height PRS is significantly associated with height. After accounting for the over-fitting implicit in identifying the best-fit PRS, the emprical p-value is 0.000999001.</p>
</details>
</blockquote>
<hr />
<h3 id="out-of-sample-validation">Out of Sample Validation</h3>
<p>The best way to avoid having results that are over-fit is to perform validation on an independent validation data set. We can perform validation of the previous height + covariate analysis with PRSice, using the independent VAL target sample as validation data and the "best" P-value threshold predicted in the VAL samples:</p>
<h2 id="rscript-softwareprsicer-prsice-softwareprsice_linux-base-base_datagiant_heighttxt-target-target_dataval-snp-markername-a1-allele1-a2-allele2-stat-b-beta-pvalue-p-pheno-target_datavalheight-binary-target-f-no-full-bar-levels-00680001-fastscore-cov-target_datavalcovariate-cov-col-sex-out-resultsheightval"><div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
    --prsice Software/PRSice_linux \
    --base  Base_Data/GIANT_Height.txt \
    --target Target_Data/VAL \
    --snp MarkerName \
    --A1 Allele1 \
    --A2 Allele2 \
    --stat b \
    --beta \
    --pvalue p \
    --pheno Target_Data/VAL.height \
    --binary-target F \
    --no-full \
    --bar-levels 0.0680001 \
    --fastscore \
    --cov Target_Data/VAL.covariate \
    --cov-col Sex \
    --out Results/Height.val
</code></pre></div></h2>
<blockquote>
<p><strong>Figure 1.4: Barplot of Height validation dataset</strong> 
<img alt="Figure 1.4" src="/images/day3/Height.val_BARPLOT_2023-06-30.png" /></p>
</blockquote>
<hr />
<hr />
<blockquote>
<p>‚ùì Why do we use --bar-levels 0.0680001 --no-full and --fastscore in this script?</p>
<p>‚ùì How does the PRS R2 and P -value for the validation data set compare to the analysis on the TAR target data? Is this what you would expect? Why?
</p>
</blockquote>
<hr />
<p><a href="#top"><a href="#table-of-contents">Back to Top</a></a></p>
<h2 id="exercise-3-distribution-of-prs">Exercise 3 Distribution of PRS</h2>
<p>Many PRS study publications include quantile plots that show an exponential increase in phenotypic value or / Odd Ratios (OR) among the top quantiles (e.g. an S-shaped quantile plot, e.g. Figure 1.6). </p>
<blockquote>
<p><strong>Figure 1.5: An example of density plot for PRS</strong></p>
<p><img alt="Figure 1.5" src="/images/day3/images020.png" /></p>
</blockquote>
<hr />
<blockquote>
<p><strong>Figure 1.6: An example of a S-shaped quantile plot</strong></p>
<p><img alt="Figure 1.6" src="/images/day3/images021.png" /></p>
</blockquote>
<hr />
<p>This might lead us to believe that individuals with PRS values in the top quantiles have a distinctly diÔ¨Äerent genetic aetiology compared to the rest of the sample, or that there is epistasis/interactions causing there substantially higher risk. However, when we plot a normally distributed variable (e.g. a PRS) as quantiles on the X-axis then we expect to observe this exponential pattern even when the X variable only has a linear eÔ¨Äect on the Y variable. This is because the top (and bottom) quantiles are further away from each other on the absolute scale of the variable and so the diÔ¨Äerences in their eÔ¨Äects are larger than between quantiles in the middle of the distribution.</p>
<p>To understand this more, we will perform a simple simulation using R:
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>R
# First, we define some simulation parameters
n.sample &lt;- 10000
PRS.r2 &lt;- 0.01
# Then, we simulate PRS that follow a random normal distribution
prs &lt;- rnorm(n.sample)
# We can then simulate the phenotype using the following script
pheno &lt;- prs + rnorm(n.sample,mean=0, sd=sqrt(var(prs)*(1-PRS.r2)/(PRS.r2)))
# We can examine the relationship between the phenotype and prs 
# using linear regression
summary(lm(pheno~prs))
# Which shows that we have the expected PRS R2
# Group the phenotype and PRS into a data.frame
info &lt;- data.frame(SampleID=1:n.sample, PRS=prs, Phenotype=pheno)
# Then we can generate the quantile plot. 
# To save time, we will load in the quantile plot script from Software
source(&quot;./Software/Quantile.R&quot;)
# Then we can plot the quantile plot using quantile_plot function
quantile_plot(info, &quot;Results/Height&quot;, 100)
</code></pre></div></p>
<blockquote>
<p><strong>Figure 1.7: The resulting quantile plot</strong></p>
<p><img alt="Figure 1.7" src="/images/day3/Height_QUANTILES_PLOT_2023-07-05.png" /></p>
</blockquote>
<hr />
<hr />
<blockquote>
<p>‚ùì What is the shape of the resulting quantile plot?</p>
<p>‚ùì Try plotting the densities of the height or CAD PRS in R * - do they look normally distributed? Why? (*Hint: You can generate a density plot for the
PRS in R using plot(density(x)) where x is a vector of the PRS values in the sample).
</p>
</blockquote>
<hr />
<p><a href="#top"><a href="#table-of-contents">Back to Top</a></a></p>
<h2 id="gene-set-analysis">Gene Set Analysis</h2>
<p>Currently, most PRS analyses have been performed on a genome-wide scale, disregarding the underlying biological pathways. Udler et al. 2018 suggest that grouping Single Nucleotide Polymorphisms (SNPs) into biological functional groups can lead to PRS that are more relevant to clinical risk. In this practical, we will go through some common file formats for gene-set analysis and will then calculate some gene-set (or pathway) based PRS.</p>
<h3 id="molecular-signatures-database-msigdb">Molecular Signatures Database MSigDB</h3>
<p>The Molecular Signatures Database (MSigDB) oÔ¨Äers an excellent source of gene-sets, including the hallmark genes, gene-sets of diÔ¨Äerent biological processes, gene-sets of diÔ¨Äerent oncogenic signatures etc. All gene-sets from MSigDB follows the Gene Matrix Transposed file format (GMT), which consists of one line per gene-set, each containing at least 3 column of data:</p>
<table>
<thead>
<tr>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">Set A</td>
<td style="text-align: center;">Description</td>
<td style="text-align: center;">Gene 1</td>
<td style="text-align: center;">Gene 2</td>
<td style="text-align: center;">...</td>
</tr>
<tr>
<td style="text-align: center;">Set A</td>
<td style="text-align: center;">Description</td>
<td style="text-align: center;">Gene 1</td>
<td style="text-align: center;">Gene 2</td>
<td style="text-align: center;">...</td>
</tr>
</tbody>
</table>
<hr />
<blockquote>
<p>** Have a look at the Reference/Sets.gmt file. **</p>
<p>‚ùì How many gene-sets are there in the Reference/Sets.gmt file? </p>
<p>‚ùì How many genes does the largest gene-set contain?
</p>
</blockquote>
<hr />
<blockquote>
<p>üí¨ While you can read the GMT file using Excel. You should be aware that Excel has a tendency to convert gene names into dates (e.g. SEPT9 to Sep-9)
</p>
</blockquote>
<hr />
<p>As GMT format does not contain the chromosomal location for each individual gene, an additional file is required to provide the chromosoaml location such that SNPs can be map to genes.</p>
<h3 id="general-transfer-format-file">General Transfer Format file</h3>
<p>The General Transfer Format (GTF) file contains the chromosomal coordinates for each gene. It is a <strong>tab</strong> separated file and all but the final field in each feature line must contain a value. "Empty" columns should be denoted with a ‚Äò.‚Äô. You can read the full format specification here. One column that might be of particular interest is column 3: <strong>feature</strong>, which indicates what feature that line of GTF represents. This allows us to select or ignore features that are of interest.</p>
<p>You can find the description of each feature <a href="http://www.sequenceontology.org/browser/obob.cgi">here</a>.</p>
<h3 id="browser-extensible-data-bed">Browser Extensible Data BED</h3>
<p>Browser Extensible Data (BED) file (diÔ¨Äerent to the binary ped file from PLINK) is a file format to define genetic regions. It contains 3 required fields per line (chromosome, start coordinate and end coordinate) together with 9 additional optional field. A special property of BED is that it is a 0-based format, i.e. chromosome starts at 0, as opposed to the usual 1-based format such as the PLINK format. For example, a SNP on chr1:10000 will be represented as:</p>
<table>
<thead>
<tr>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;"><strong>1</strong></td>
<td style="text-align: center;"><strong>9999</strong></td>
<td style="text-align: center;"><strong>10000</strong></td>
</tr>
</tbody>
</table>
<hr />
<blockquote>
<p>‚ùì How should we represent the coordinate of rs2980300 (chr1:785989) in BED format?
</p>
</blockquote>
<hr />
<p><a href="#top"><a href="#table-of-contents">Back to Top</a></a></p>
<h2 id="gene-set-enrichment-analysis">Gene Set Enrichment Analysis</h2>
<p>Now we have gone through all the files involved in gene-set analysis, we should consider one of the most important aspects of gene-set (or pathway) enrichment
analyses, which is the diÔ¨Äerent types of testing that we can perform when doing them:</p>
<h3 id="self-contained-vs-competitive-testing">Self-Contained vs Competitive Testing</h3>
<p>The null-hypothesis of self-contained and competitive test statistics is diÔ¨Äerent:
  ‚Äì <strong>Self-Contained</strong> - None of the genes within the gene-set are associated with the phenotype
  ‚Äì <strong>Competitive</strong> - Genes within the gene-set are no more associated with the phenotype than genes outside the gene-set
Therefore, a bigger gene-set will have a higher likelihood of having a significant P -value from self-contained test, which is not desirable.</p>
<h2 id="exercise-4-gene-set-based-prs-analysis">Exercise 4 Gene Set Based PRS Analysis</h2>
<p>Having learnt about the basics of gene-set analyses, we are now ready to perform gene-set association analyses using PRSet.</p>
<p>To perform the PRSet analysis and obtain the set based PRS and competitive P-value, simply provide the GTF file and the GMT file to PRSice and specify the number of permutation for competitive P-value calculation using the --set-perm option.</p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>Rscript ./Software/PRSice.R \
    --prsice Software/PRSice_linux  \
    --base Base_Data/GIANT_Height.txt \
    --target Target_Data/TAR \
    --A1 Allele1 \
    --A2 Allele2 \
    --snp MarkerName \
    --pvalue p \
    --stat b \
    --beta \
    --binary-target F \
    --pheno Target_Data/TAR.height \
    --cov Target_Data/TAR.covariate \
    --out Results/Height.set \
    --gtf Reference/Homo_sapiens.GRCh38.86.gtf \
    --wind-5 5kb \
    --wind-3 1kb \
    --msigdb Reference/Sets.gmt \
    --multi-plot 10 \
    --set-perm 1000
</code></pre></div>
<blockquote>
<p><strong>Figure 1.8: An example of the multi-set plot. Sets are sorted based on their self-contained R2 . Base is the genome wide PRS</strong></p>
<p><img alt="Figure 1.8" src="/images/day3/Height.set_MULTISET_BARPLOT_2023-06-30.png" /></p>
</blockquote>
<hr />
<blockquote>
<p>üìå If the --wind-5 and --wind-3 flag is not specified, PRSet will use the exact coordinates of each gene as the boundary. By specifying eg. --wind-5 5kb and --wind-3 1kb then the boundary of each gene will be extended 5 kb towards the 5‚Äô end and 1 kb towards the 3‚Äô end so that regulatory elements of the gene can be included.</p>
<p>üìç By default, when calculating set based PRS, PRSet will not perform P -value thresholding. This is because the aim in gene-set analyses is to assess the overall signal in each gene-set, and compare which is most enriched for signal, rather than optimise predictive power as typically desirable for genome-wide PRS. Providing any of the following commands will activate P -value thresholding for set based PRS calculation: --lower, --upper, --inter, --bar-levels, --fastscore
</p>
</blockquote>
<hr />
<blockquote>
<p>‚ùì Can you plot the relationship between the gene-set R2 and the number of SNPs in each gene-set? What general trend can be seen?</p>
<p>‚ùì Considering the plot, what gene-sets do you think are most interesting and why?</p>
<p>‚ùì Why is it useful to have polygenic scores measured across gene-sets (or pathways) for individuals? Isn‚Äôt it suÔ¨Écient to just obtain a ranking of gene-sets according to GWAS-signal enrichment?
</p>
</blockquote>
<hr />
              
            </div>
          </div><footer>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/tadesouaiaia/sibArc-website" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
    
  </span>
</div>
    <script src="../../../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../../../..";</script>
    <script src="../../../../js/theme_extra.js"></script>
    <script src="../../../../js/theme.js"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      <script src="../../../../javascripts/details.js"></script>
      <script src="../../../../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
